---
title: "Mykrobe_AMRconcordance_2024"
author: "DanielleIngle"
date: "18/01/2024 - 19/010/2024"
output: html_document
---

```{r setup, include=FALSE}
require('knitr')
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = getwd())
getwd()

```

```{r library}

library(tidyverse)
library(readxl)
library(ggplot2)
library(patchwork)
library(RColorBrewer)
library(ggsankey)

sessionInfo()
```


```{r files}
#list of isolates included 
gtcalls.tgc <- read_csv("TGC_Genotypes_3methods_n12839_28112023.csv")
#for linking with PW
tgc.id <- read_csv("/Users/daniellei/Documents/github/TyphoidGenomicsConsortiumWG2/TGC_data.csv")


#AMR details from the three methods
bam.amr <- read_csv("GenoTyphi_PythonBam_n13032_28112023.csv")
myk.amr <- read_csv("Genotyphi_MYKROBE_merge_n13033_28112023.csv")
pw.amr <- read_csv("../PublicData/PWdata_2023/PW_TyphiNET-DB_plus_Rwanda.csv")
```



```{r into one file}
#add 'Bam' or 'Myk' or "PW' to each fo the three files so know where AMR comes from
#append col names

colnames(bam.amr) <- paste("BAM", colnames(bam.amr), sep = "_")
colnames(myk.amr) <- paste("MYK", colnames(myk.amr), sep = "_")
colnames(pw.amr) <- paste("PW", colnames(pw.amr), sep = "_")

#add to gtcalls.tgc
# sort out the point mutations 
amr.deets.bam <- gtcalls.tgc %>% left_join(bam.amr, by = c("Sanger_ID" = "BAM_Sanger_ID")) %>% mutate(`BAM_gyrA-S83F` = str_extract(`BAM_AMR mutations`, "gyrA-S83F")) %>% mutate(`BAM_gyrA-S83Y` = str_extract(`BAM_AMR mutations`, "gyrA-S83Y")) %>% mutate(`BAM_gyrA-D87G` = str_extract(`BAM_AMR mutations`, "gyrA-D87G")) %>% mutate(`BAM_gyrA-D87N` = str_extract(`BAM_AMR mutations`, "gyrA-D87N")) %>% mutate(`BAM_gyrA-D87V` = str_extract(`BAM_AMR mutations`, "gyrA-D87V")) %>% mutate(`BAM_gyrA-D87Y` = str_extract(`BAM_AMR mutations`, "gyrA-D87Y")) %>% mutate(`BAM_gyrB-S464Y` = str_extract(`BAM_AMR mutations`, "gyrB-S464Y"))  %>% mutate(`BAM_gyrB-S464F` = str_extract(`BAM_AMR mutations`, "gyrB-S464F")) %>% mutate(`BAM_parC-S80I` = str_extract(`BAM_AMR mutations`, "parC-S80I")) %>% mutate(`BAM_parC-E84K` = str_extract(`BAM_AMR mutations`, "parC-E84K")) %>% mutate(`BAM_parC-S80R` = str_extract(`BAM_AMR mutations`, "parC-S80R")) %>% mutate(`BAM_parC-E84G` = str_extract(`BAM_AMR mutations`, "parC-E84G")) %>% mutate(`BAM_acrB-R717L` = str_extract(`BAM_AMR mutations`, "acrB-R717L")) %>% mutate(`BAM_acrB-R717Q` = str_extract(`BAM_AMR mutations`, "acrB-R717Q"))

#add in myk calls
amr.deets.bamyk <- amr.deets.bam %>% left_join(myk.amr, by = c("Sanger_ID" = "MYK_Sanger_ID")) 

#add in PW calls
# use tgc.id$PW_id and tgc.pw$`Genome ID` to merge
pw.tgc.ids<- tgc.id[,c(1:18,107:108)] %>% left_join(pw.amr, by=c("PW_id" = "PW_Genome ID"))

## use tgc.id$PW_id and tgc.pw$`Genome ID` to merge
#remove the three duplicates
pw.tgc.ids<- tgc.id[,c(1,107:108)] %>% left_join(pw.amr, by=c("PW_id" = "PW_Genome ID")) 
pw.tgc.ids$TGC_sangerlane[duplicated(pw.tgc.ids$TGC_sangerlane)]
pw.tgc.ids <- distinct(pw.tgc.ids, TGC_sangerlane, .keep_all = TRUE)

amr.deets.fin <- amr.deets.bamyk %>% left_join(pw.tgc.ids[,c(1,31:51,60:97,101)], by = "TGC_sangerlane") 


write_csv(amr.deets.fin, "AMR_Bam_Myk_PW_fulldata_18012024_n12839.csv")
```

```{r filter to 11992}

#those with PW calls and mykrobe typhi calls
table(amr.deets.fin$`PW_Species Name`, useNA = 'ifany')
table(amr.deets.fin$MYK_species, useNA = 'ifany')

amr.data <- amr.deets.fin %>% filter(`PW_Species Name` == "Salmonella enterica") %>% filter(MYK_species == "typhi")

paste("Final number of isolates for AMR comparison = ", nrow(amr.data), sep ="")

write_csv(amr.data, "Mykrobe_AMRconcordance_18012024_n11992.csv")
```


```{r CEF}
#not point mutations for CEF - only Myk and PW
amr.data %>% group_by(`MYK_ceftriaxone`,PW_ESBL_category) %>% summarise(n=n())

cef.data <- amr.data %>% filter(MYK_ceftriaxone != "S" | PW_ESBL_category != "Non-ESBL" )
cef.data %>% group_by(`MYK_ceftriaxone`,PW_ESBL_category) %>% summarise(n=n())

#need to gather from wide to long the isolates with any one of the four mechanisms from PW
cef.data$PW_Ceftriaxone <- ifelse(cef.data$`PW_blaCTX-M-15_23` == 1, "blaCTX-M-15", ifelse(cef.data$`PW_blaSHV-12` == 1, "blaSHV-12",ifelse(cef.data$`PW_blaCTX-M-55` == 1, "blaCTX-M-55",ifelse(cef.data$`PW_blaCTX-M-12` == 1, "blaCTX-M-12",ifelse(cef.data$`PW_blaOXA-7` == 1, "blaOXA-7",ifelse(cef.data$`PW_blaOXA134_2` == 1, "blaOXA-134",ifelse(cef.data$`PW_ampC` == 1, "ampC","S")))))))

#note that the two single isolates with ctx-m-12 adn ctx-m-55 - not have the ctx-m-15 or blaSHV12
cef.data %>% group_by(`MYK_ceftriaxone`,PW_Ceftriaxone) %>% summarise(n=n())

cef.data.df <- cef.data[,c(1,12,75,114,118,192,197,159:161,163:165)]

paste("Number of isolates with Cef resistance from either Myk or PW = ", nrow(cef.data), sep = "")
paste("Number of isolates that matched as susceptible = ",nrow(amr.data) - nrow(cef.data), sep = "")

paste("Total percentage of ", nrow(cef.data), " CEF R isolates that matched CEF mechanisms ",(nrow(cef.data)-4) /nrow(cef.data)*100, sep = "")
paste("Total percentage of isolates that matched as Cef resistance ",(nrow(cef.data)-4) /nrow(cef.data)*100, sep = "")

cef.data.df$CEF <- "Resistance"
cef.gene <- cef.data %>% group_by(`MYK_ceftriaxone`,PW_Ceftriaxone) %>% summarise(n=n())

write_csv(cef.data.df,"AMRconcordance_CEF_19012024_n693.csv")
write_csv(cef.gene,"AMRconcordance_CEF_19012024_PW_MYK_n693.csv")


df.esbl <- cef.data.df %>% make_long(CEF, MYK_ceftriaxone, PW_Ceftriaxone)
head(df.esbl)

# Chart 1 - ESBL Resistance only
pl.esbl <- ggplot(df.esbl, aes(x = x
                     , next_x = next_x
                     , node = node
                     , next_node = next_node
                     , fill = factor(node)
                     , label = node)
             ) # + geom_sankey()
pl.esbl <- pl.esbl +geom_sankey(flow.alpha = 0.5
                      , node.color = "black"
                      ,show.legend = FALSE)
pl.esbl <- pl.esbl +geom_sankey_label(size = 3, color = "black", fill= "white", hjust = -0.15)
pl.esbl <- pl.esbl +  theme_bw()
pl.esbl <- pl.esbl + theme(legend.position = "none")
pl.esbl <- pl.esbl +  theme(axis.title = element_blank()
                  , axis.text.y = element_blank()
                  , axis.ticks = element_blank()  
                  , panel.grid = element_blank())
pl.esbl <- pl.esbl +  theme(panel.border = element_blank())
pl.esbl <- pl.esbl + scale_fill_viridis_d(option = "viridis")
pl.esbl <- pl.esbl + labs(title = "Ceftriaxone Resistance \n n = 693")
pl.esbl <- pl.esbl + labs(fill = 'Nodes')
pl.esbl


```




```{r AZI}
#point mutations only
amr.data %>% group_by(MYK_azithromycin,PW_acrB_R717Q,PW_acrB_R717L,`BAM_acrB-R717L`, `BAM_acrB-R717Q`) %>% summarise(n=n())

azi.data <- amr.data %>% filter(MYK_azithromycin != "S"  ) %>% replace_na(list("BAM_acrB-R717L" =0)) %>% replace_na(list("BAM_acrB-R717Q" =0))

azi.data %>% group_by(MYK_azithromycin,PW_acrB_R717Q,PW_acrB_R717L,`BAM_acrB-R717L`, `BAM_acrB-R717Q`) %>% summarise(n=n())

#need to gather from wide to long the isolates with any one of the four mechanisms from PW
azi.data$PW_Azithromycin <- ifelse(azi.data$`PW_acrB_R717Q` == 1, "acrB-R717Q", ifelse(azi.data$`PW_acrB_R717L` == 1, "acrB-R717L","S"))

azi.data$Mapping_Azithromycin <- ifelse(azi.data$`BAM_acrB-R717L` == "acrB-R717L", "acrB-R717L", ifelse(azi.data$`BAM_acrB-R717Q` == "acrB-R717Q", "acrB-R717Q","S"))

#only known point mutations
azi.data %>% group_by(`MYK_azithromycin`,PW_Azithromycin,Mapping_Azithromycin) %>% summarise(n=n())

azi.data.df <- azi.data[,c(1,12,48,61:62,198,74,108:112,153,197,149:150)]

paste("Number of isolates with AZI resistance from either Myk or PW or BAM = ", nrow(azi.data), sep = "")
paste("Number of isolates that matched as susceptible for AZI = ",nrow(amr.data) - nrow(azi.data), sep = "")

paste("Total percentage of ", nrow(azi.data), " AZI R isolates that matched AZI mechanisms ",(nrow(azi.data)-1) /nrow(azi.data)*100, sep = "")

azi.data.df$AZI <- "Resistance"
azi.data.df %>% group_by(MYK_azithromycin, Mapping_Azithromycin)  %>%   summarise(n=n())
#BAM and Mykrobe match exactly
azi.gene <-  azi.data.df %>% group_by(MYK_azithromycin, PW_Azithromycin)  %>%   summarise(n=n())

write_csv(azi.data.df,"AMRconcordance_AZI_19012024_mechanism_n74.csv")
write_csv(azi.gene,"AMRconcordance_MYK_PW_AZI_19012024_n74.csv")


df.azi <- azi.data.df %>% make_long(AZI, MYK_azithromycin, Mapping_Azithromycin, PW_Azithromycin)
head(df.azi)

# Chart - AZI Resistance only
pl.azi <- ggplot(df.azi, aes(x = x
                     , next_x = next_x
                     , node = node
                     , next_node = next_node
                     , fill = factor(node)
                     , label = node)
             )
pl.azi <- pl.azi +geom_sankey(flow.alpha = 0.5
                      , node.color = "black"
                      ,show.legend = FALSE)
pl.azi <- pl.azi +geom_sankey_label(size = 3, color = "black", fill= "white", hjust = -0.15)
pl.azi <- pl.azi +  theme_bw()
pl.azi <- pl.azi + theme(legend.position = "none")
pl.azi <- pl.azi +  theme(axis.title = element_blank()
                  , axis.text.y = element_blank()
                  , axis.ticks = element_blank()  
                  , panel.grid = element_blank())
pl.azi <- pl.azi +  theme(panel.border = element_blank())
pl.azi <- pl.azi + scale_fill_viridis_d(option = "viridis")
pl.azi <- pl.azi + labs(title = "Azithromycin Resistance \nn = 74")
pl.azi <- pl.azi + labs(fill = 'Nodes')
pl.azi

```


```{r CIP}
#PW CIP high level summary
amr.data %>% group_by(PW_dcs_category,PW_cip_pred_pheno,PW_cip_pheno_qrdr_gene)  %>% summarise(n=n())

#point mutations only
amr.data %>% group_by(MYK_ciprofloxacin) %>% summarise(n=n())
amr.data %>% group_by(PW_cip_pred_pheno) %>% summarise(n=n())
# difference of n = 3573 for Myk S and 3578 for PW S. n = 3624 for BAM S (point mutations only)
table(amr.data$`BAM_AMR mutations`)
table(amr.data$`BAM_AMR mutations`, useNA = "ifany")

#qnr for PW
amr.data %>% group_by(PW_cip_pheno_qrdr_gene) %>% summarise(n=n())

#overview of R I or S
amr.data.cip <- amr.data[,c(1,12,48:60,76,95:107,151,155:157,137:148, 169:171)] %>% separate(col = "MYK_ciprofloxacin", sep = ":", c("MYK RIS","Myk Cip Geno Mechanism"), remove = FALSE, fill = "right")

#filter to those 
amr.data.cip <-  amr.data.cip %>% replace_na(list("BAM_AMR mutations" = "S" )) %>%  mutate(CIP =ifelse(MYK_ciprofloxacin != "S"   | PW_cip_pheno_qrdr_gene != "CipS00" | `BAM_AMR mutations` != "S", "Resistance", "Susceptible"))

cip.data <- amr.data.cip %>% filter(CIP == "Resistance")


paste("Number of isolates with CIP resistance from either Myk or PW or BAM = ", nrow(cip.data), sep = "")
paste("Number of isolates that matched as susceptible for CIP = ",nrow(amr.data.cip)-nrow(cip.data) , sep = "")


df.cip <- cip.data %>% make_long(CIP,`MYK RIS`, PW_cip_pred_pheno )

# Chart - CIP Resistance only
pl.cip <- ggplot(df.cip, aes(x = x
                     , next_x = next_x
                     , node = node
                     , next_node = next_node
                     , fill = factor(node)
                     , label = node)
             )
pl.cip <- pl.cip +geom_sankey(flow.alpha = 0.5
                      , node.color = "black"
                      ,show.legend = FALSE)
pl.cip <- pl.cip +geom_sankey_label(size = 3, color = "black", fill= "white", hjust = -0.15)
pl.cip <- pl.cip +  theme_bw()
pl.cip <- pl.cip + theme(legend.position = "none") +  theme(axis.title = element_blank()
                  , axis.text.y = element_blank()
                  , axis.ticks = element_blank()  
                  , panel.grid = element_blank()) +  theme(panel.border = element_blank())
pl.cip <- pl.cip + scale_fill_viridis_d(option = "viridis")
pl.cip <- pl.cip + labs(title = "Ciprofloxacin Resistance \nn = 8423")
pl.cip <- pl.cip + labs(fill = 'Nodes')
pl.cip


#need to gather from wide to long the isolates with mechanisms in single col for PW and BAM. In same order of genes as Mykrobe.
#unite the values for the BAM data into a single col

cip.data.mech<- amr.data.cip %>% unite(Mapping_Cip, 4:15, remove = FALSE,na.rm = TRUE, sep = ";")  %>% mutate(Mapping_Cipfloxacin = ifelse(Mapping_Cip == "", "S", Mapping_Cip))  
#table by the profiles - number detected for each of Mykrobe and PW as have qnr genes as well 
cip.mech <- cip.data.mech %>% group_by(Mapping_Cipfloxacin, MYK_ciprofloxacin)  %>%   summarise(n=n())

write_csv(cip.mech, "AMRgenes_Myk_BAM_CIP_19012024.csv")
```

```{r MDR}
#PW MDR high level summary
amr.data %>% group_by(PW_MDR)  %>% summarise(n=n())

#filter to those 
amr.data.mdr <-  amr.data %>%  mutate(MYK_MDR =ifelse(MYK_ampicillin != "S"   & MYK_chloramphenicol != "S" & MYK_sulfamethoxazole != "S", "MDR", "Susceptible"))
amr.data.mdr %>% group_by(MYK_MDR,PW_MDR)  %>% summarise(n=n())
#filter and add cols for plot
mdr.data <- amr.data.mdr %>% filter(MYK_MDR != "Susceptible" | PW_MDR != "-" ) %>% separate(col = "MYK_ampicillin", sep = ":", c("MYK AMP","Myk AMP Mechanism"), remove = FALSE, fill = "right") %>% separate(col = "MYK_chloramphenicol", sep = ":", c("MYK CAT","Myk CAT Mechanism"), remove = FALSE, fill = "right") %>% separate(col = "MYK_sulfamethoxazole", sep = ":", c("MYK CoTrim","Myk CoTrim Mechanism"), remove = FALSE, fill = "right")

mdr.data %>% group_by(`MYK AMP`,`PW_blaTEM-1D`)  %>% summarise(n=n())
mdr.data %>% group_by(`MYK CAT`,PW_chloramphenicol_category)  %>% summarise(n=n())
mdr.data %>% group_by(`MYK CoTrim`,PW_co_trim)  %>% summarise(n=n())



paste("Number of isolates with MDR resistance from either Myk or PW  = ", nrow(mdr.data), sep = "")

mdr.data$MDR <- "Resistance"


df.mdr <- mdr.data %>% make_long(MDR,MYK_MDR, PW_MDR, `MYK AMP`,`PW_blaTEM-1D`,`MYK CAT`, PW_chloramphenicol_category,`MYK CoTrim`,PW_co_trim)

# Chart - MDR Resistance only
pl.mdr <- ggplot(df.mdr, aes(x = x
                     , next_x = next_x
                     , node = node
                     , next_node = next_node
                     , fill = factor(node)
                     , label = node)
             )
pl.mdr <- pl.mdr +geom_sankey(flow.alpha = 0.5
                      , node.color = "black"
                      ,show.legend = FALSE)
pl.mdr <- pl.mdr +geom_sankey_label(size = 3, color = "black", fill= "white", hjust = -0.15)
pl.mdr <- pl.mdr +  theme_bw()
pl.mdr <- pl.mdr + theme(legend.position = "none") +  theme(axis.title = element_blank()
                  , axis.text.y = element_blank()
                  , axis.ticks = element_blank()  
                  , panel.grid = element_blank()) +  theme(panel.border = element_blank())
pl.mdr <- pl.mdr + scale_fill_viridis_d(option = "magma")
pl.mdr <- pl.mdr + labs(title = "MDR Resistance \nn = 3383")
pl.mdr <- pl.mdr + labs(fill = 'Nodes')
pl.mdr


```


```{r AMR profiles}

#only one ampicillin one - rest bla genes for CEF
amr.data %>% group_by(MYK_ampicillin, `PW_blaTEM-1D`)  %>% summarise(n=n())
amr.data.amp <- amr.data %>% mutate(PW_Ampicillin = ifelse(`PW_blaTEM-1D` == 1, "blaTEM-1D", "S"))  
amp.gene <- amr.data.amp %>% group_by(MYK_ampicillin, PW_Ampicillin)  %>% summarise(n=n())

#CAT
amr.data %>% group_by(MYK_chloramphenicol, PW_chloramphenicol_category)  %>% summarise(n=n())
amr.data %>% group_by(MYK_chloramphenicol, PW_catA1, PW_cmlA)  %>% summarise(n=n())

amr.data.cat <- amr.data %>% mutate(PW_cat1_gene = ifelse(PW_catA1 == 1, "catA1", NA))  %>% mutate(PW_cmlA_gene = ifelse(PW_cmlA == 1, "cmlA", NA))  %>% unite(PW_CAT_genes, PW_cat1_gene:PW_cmlA_gene, remove = FALSE,na.rm = TRUE, sep = ";") %>% mutate(PW_Chloramphenicol = ifelse(PW_CAT_genes == "", "S", PW_CAT_genes)) 
cat.gene <- amr.data.cat %>% group_by(MYK_chloramphenicol, PW_Chloramphenicol)  %>%   summarise(n=n())


#CO-TRIM
amr.data %>% group_by(MYK_sulfamethoxazole, PW_co_trim)  %>% summarise(n=n())

#SUl
amr.data %>% group_by(MYK_sulfonamides, PW_sul_any)  %>% summarise(n=n())
amr.data %>% group_by(MYK_sulfonamides, PW_sul1, PW_sul2)  %>% summarise(n=n())

amr.data.sul <- amr.data %>% mutate(PW_sul1_gene = ifelse(PW_sul1 == 1, "sul1", NA))  %>% mutate(PW_sul2_gene = ifelse(PW_sul2 == 1, "sul2", NA))  %>% unite(PW_Sulf_genes, PW_sul1_gene:PW_sul2_gene, remove = FALSE,na.rm = TRUE, sep = ";") %>% mutate(PW_Sulfonomides = ifelse(PW_Sulf_genes == "", "S", PW_Sulf_genes)) 
sul.gene <- amr.data.sul %>% group_by(MYK_sulfonamides, PW_Sulfonomides)  %>%   summarise(n=n())


#DFR
amr.data %>% group_by(MYK_trimethoprim, PW_dfra_any)  %>% summarise(n=n())

amr.data.dfr <- amr.data %>% mutate(PW_dfrA1_gene = ifelse(PW_dfrA1 == 1, "dfrA1", NA))  %>% mutate(PW_dfrA14_gene = ifelse(PW_dfrA14 == 1, "dfrA14", NA))  %>% mutate(PW_dfrA15_gene = ifelse(PW_dfrA15 == 1, "dfrA15", NA))  %>% mutate(PW_dfrA17_gene = ifelse(PW_dfrA17 == 1, "dfrA17", NA))  %>% mutate(PW_dfrA18_gene = ifelse(PW_dfrA18 == 1, "dfrA18", NA)) %>% mutate(PW_dfrA5_gene = ifelse(PW_dfrA5 == 1, "dfrA5", NA))  %>% mutate(PW_dfrA7_gene = ifelse(PW_dfrA7 == 1, "dfrA7", NA)) %>% unite(PW_Trim_genes, PW_dfrA1_gene:PW_dfrA7_gene, remove = FALSE,na.rm = TRUE, sep = ";") %>% mutate(PW_Trimethoprim = ifelse(PW_Trim_genes == "", "S", PW_Trim_genes)) 
dfr.gene <- amr.data.dfr %>% group_by(MYK_trimethoprim, PW_Trimethoprim)  %>%   summarise(n=n())


#tet 
amr.data %>% group_by(MYK_tetracycline, PW_tetracycline_category)  %>% summarise(n=n())
amr.data %>% group_by(MYK_tetracycline, `PW_tetA(A)`, `PW_tetA(B)`, `PW_tetA(C)`, `PW_tetA(D)`)  %>% summarise(n=n())

amr.data.tet <- amr.data %>% mutate(PW_tetA_gene = ifelse(`PW_tetA(A)` == 1, "tetA", NA))  %>% mutate(PW_tetB_gene = ifelse(`PW_tetA(B)` == 1, "tetB", NA)) %>% mutate(PW_tetC_gene = ifelse(`PW_tetA(C)` == 1, "tetC", NA))  %>% mutate(PW_tetD_gene = ifelse(`PW_tetA(D)` == 1, "tetD", NA)) %>% unite(PW_TET_genes, PW_tetA_gene:PW_tetD_gene, remove = FALSE,na.rm = TRUE, sep = ";") %>% mutate(PW_Tetracycline = ifelse(PW_TET_genes == "", "S", PW_TET_genes)) 
tet.gene <- amr.data.tet %>% group_by(MYK_tetracycline, PW_Tetracycline)  %>%   summarise(n=n())


#write out
write_csv(amp.gene, "AMRgenes_Myk_PW_AMP_19012024.csv")
write_csv(cat.gene, "AMRgenes_Myk_PW_CAT_19012024.csv")
write_csv(sul.gene, "AMRgenes_Myk_PW_SUL_19012024.csv")
write_csv(dfr.gene, "AMRgenes_Myk_PW_DFR_19012024.csv")
write_csv(tet.gene, "AMRgenes_Myk_PW_TET_19012024.csv")

```


```{r AMR all}
#AMR except CIP in one table
cef.gene$`Drug class` <- "Ceftriaxone"
cef.gene <- as.data.frame(cef.gene) %>% rename("Mykrobe" = MYK_ceftriaxone, "PathogenWatch" = PW_Ceftriaxone)

azi.gene$`Drug class` <- "Azithromycin"
azi.gene <- as.data.frame(azi.gene) %>% rename("Mykrobe" = MYK_azithromycin, "PathogenWatch" = PW_Azithromycin)

amp.gene$`Drug class` <- "Ampicillin"
amp.gene <- as.data.frame(amp.gene) %>% rename("Mykrobe" = MYK_ampicillin, "PathogenWatch" = `PW_Ampicillin`)

cat.gene$`Drug class` <- "Cholramophenicol"
cat.gene <- as.data.frame(cat.gene) %>% rename("Mykrobe" = MYK_chloramphenicol, "PathogenWatch" = `PW_Chloramphenicol`)

dfr.gene$`Drug class` <- "Trimethoprim"
dfr.gene <- as.data.frame(dfr.gene) %>% rename("Mykrobe" = MYK_trimethoprim, "PathogenWatch" = `PW_Trimethoprim`)

sul.gene$`Drug class` <- "Sulfonamides"
sul.gene <- as.data.frame(sul.gene) %>% rename("Mykrobe" = MYK_sulfonamides, "PathogenWatch" = `PW_Sulfonomides`)

drugs.calls <- rbind(cef.gene, azi.gene, amp.gene, cat.gene, dfr.gene,sul.gene)
drugs.df <- drugs.calls[,c(4,1:3)]

write_csv(drugs.df, "AMRgenes_Myk_PW_allclasses_19012024.csv")

```


```{r pdf image}


pl.mdr + pl.cip +  plot_layout(widths = c(2, 1))

pl.esbl | pl.azi

(pl.mdr + pl.cip +  plot_layout(widths = c(2, 1))) /
  (pl.esbl + pl.azi)


pdf("AMRgenePWMyk_concordance_Sankey_MDR_CIP_19012024.pdf")
pl.mdr + pl.cip +  plot_layout(widths = c(2, 1))
dev.off()

pdf("AMRgenePWMyk_concordance_Sankey_ESBL_AZI_19012024.pdf")
pl.esbl | pl.azi
dev.off()

pdf("AMRgenePWMyk_concordance_Sankey_4panel_19012024.pdf")
(pl.mdr + pl.cip +  plot_layout(widths = c(2, 1))) /
  (pl.esbl + pl.azi)
dev.off()

```  

