---
title: "Mykrobe_AMR_PhenoGenoData_2024"
author: "DanielleIngle"
date: "12/01/2024 - 15/01/2024"
output: html_document
---

```{r setup, include=FALSE}
require('knitr')
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = getwd())
getwd()

```

```{r library}

library(tidyverse)
library(ggplot2)
library(readxl)
library(RColorBrewer)
library(ComplexUpset)
library(scales)
library(patchwork)

sessionInfo()
```



```{r files }

#List of isolates that passed QC and has Mykrobe and Bam data (and Typhi)
#n= 12839
gtcalls.tgc <- read_csv("TGC_Genotypes_3methods_n12839_28112023.csv")

#final Mykrobe data full. Will filter to those in table above
myk.full.df<-  read_csv("Genotyphi_MYKROBE_merge_n13033_28112023.csv")

#public datasets
#UKHSA supplementary table with MIC
uk.mic <- read_excel("../PublicData/mgen-7-0633-s002.xlsx", sheet = 3)

#SEAP study supplementary with Disc Diffusion data. Some warnings with dates - not important
seap.dd <- read_excel("../PublicData/mmc2.xlsx")

#CDC data with MIC data in public domain - https://wwwn.cdc.gov/narmsnow/
cdc.mic <- read_excel("../PublicData/GTGC-phenotype-genotype-comparison-NARMSData-040423.xlsx", sheet = 1)

```

```{r details of Mykr with cols needed to merge}

myk.details.data <- gtcalls.tgc[,c(1,12:18,22:23)] %>% left_join(myk.full.df,by = c("Sanger_ID", "genome"))
nrow(myk.details.data )

myk.details.data  %>% group_by(species) %>% summarize(n = n()) 
```


```{r UK merge}
#check have Typhi isolates
uk.mic  %>% group_by(`Organism Identified`) %>% summarize(n = n()) 

#add dataset info/ source
uk.mic$DataSource <- "UKHSA MGen 2021"


#left join with the main myk data and then filter to UK isolates
myk.data.uk.mic <- myk.details.data %>%left_join(uk.mic, by = c("Sanger_ID" =  "Accession Number"))
myk.data.uk.mic %>% group_by(DataSource) %>% summarise(n=n())
myk.data.uk.mic <- myk.data.uk.mic %>% filter(DataSource == "UKHSA MGen 2021")

#write out
write_csv(myk.data.uk.mic, "Myk_UKHSA_MIC_15012024_n852.csv")
```


```{r SEAP merge}

#add dataset info/ source
seap.dd$DataSource <- "SEAP LanMicrobe 2022"

#left join with the main myk data and then filter to UK isolates
myk.data.seap.dd <- myk.details.data %>%left_join(seap.dd, by = c("Sanger_ID" =  "Genome name"))
myk.data.seap.dd %>% group_by(DataSource) %>% summarise(n=n())
myk.data.seap.dd <- myk.data.seap.dd %>% filter(DataSource == "SEAP LanMicrobe 2022")

#write out
write_csv(myk.data.seap.dd, "Myk_SEAP_DD_15012024_n2446.csv")

```


```{r CDC merge}

#add dataset info/ source
cdc.mic$DataSource <- "CDC MIC Public"
nrow(cdc.mic)

#remove 'NotUploaded'
cdc.mic.up <- cdc.mic %>% filter(`NCBI Accession Number` != "NotUploaded")
cdc.mic.not <- cdc.mic %>% filter(`NCBI Accession Number` == "NotUploaded")
paste("Number of isolates where data not uploaded = ",nrow(cdc.mic.not), sep ="")


#id duplicates
cdc.mic.up$`WGS ID`[duplicated(cdc.mic.up$`WGS ID`)]
paste("Number of isolates iwth duplicates = ", length(cdc.mic.up$`WGS ID`[duplicated(cdc.mic.up$`WGS ID`)]), sep ="")
nrow(cdc.mic.up)

#dataset, distinct values on Display name keep all cols
#Removed the 32 duplicates
#https://dplyr.tidyverse.org/reference/distinct.html
cdc.mic.uni <- distinct(cdc.mic.up, `WGS ID`, .keep_all = TRUE)
nrow(cdc.mic.uni)


#WGS ID in cdc table has the PNUS ids - match the DisplayName col
#NCBI Accession Number has the SAMN - marches the Sample Accession
#left join with the main myk data and then filter to CDC isolates
#need to filter out the ones that have display name == NA -these link to other isolates in the dataset 
myk.data.cdc.mic <- myk.details.data %>%left_join(cdc.mic.uni, by = c("DisplayName" = "WGS ID", "Sample Accession" = "NCBI Accession Number" ))
myk.data.cdc.mic %>% group_by(DataSource) %>% summarise(n=n())
myk.data.cdc.mic <- myk.data.cdc.mic %>% filter(DataSource == "CDC MIC Public")
nrow(myk.data.cdc.mic)

#write out
write_csv(myk.data.cdc.mic, "Myk_CDC_MIC_15012024_n720.csv")

```



```{r UK upset CIP}

cip.mech.list <- rev(colnames(myk.data.uk.mic)[c(47:54,43:46,66:68)])


#the < and > for some of the numbers are problematics
#transform the CIP_MIC_edit to numeric - needed for plotting - currently in character. 
#ok to change as can only go up or down for the two
myk.data.uk.mic.cip <- myk.data.uk.mic %>%  mutate(CIP_MIC_edit =ifelse(`CIP - Ciprofloxacin` == "<0.015", "0.0075",ifelse(`CIP - Ciprofloxacin` == ">8", "16",ifelse(`CIP - Ciprofloxacin` == "0.06", "0.064",`CIP - Ciprofloxacin`)))) %>% transform(CIP_MIC_edit = as.numeric(CIP_MIC_edit) )%>% separate(col = "ciprofloxacin", sep = ":", c("CIP","Cip Geno Mechanism"), remove = FALSE, fill = "right")

#upset plots
#need to change the tick marks on the y axis for violin plot
#add break 0.03 - is a breakpoint just not reported here. 
#jitter for individual plots

upset(myk.data.uk.mic.cip,
      cip.mech.list,
      sort_sets=FALSE,
      base_annotations=list('Intersection size'=intersection_size(counts=TRUE,mapping=aes(fill=`CIP`)) + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26','I'='#fc9272'))),
      annotations = list(
          "MIC"=ggplot(mapping=aes(x=intersection, y=CIP_MIC_edit)) + ggtitle("CIP mechanisms v MIC") + geom_jitter(alpha=0.2) + geom_violin(aes(fill = `CIP`),width=1, alpha=1)  + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26','I'='#fc9272')) +  scale_y_continuous(trans=log2_trans(), breaks=c(0.0075,0.015,0.032, 0.064,0.125,0.25, 0.5,1,2,4,8, 16))))


#fig for paper
upset(myk.data.uk.mic.cip,
      cip.mech.list,
      sort_sets=FALSE,
      width_ratio=0.1,
      set_sizes=(
          upset_set_size(position='right')
          + theme(axis.text.x=element_text(angle=90))
          ),
      guides='over',
      base_annotations=list('Intersection size'=intersection_size(counts=TRUE,mapping=aes(fill=`CIP`)) + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26','I'='#fc9272'))),
      annotations = list(
          "MIC"=ggplot(mapping=aes(x=intersection, y=CIP_MIC_edit)) + ggtitle("UKHSA \nCIP mechanisms v MIC")  + geom_violin(aes(fill = `CIP`),width=1, alpha=1)  + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26','I'='#fc9272'), guide = 'none') +  scale_y_continuous(trans=log2_trans(), breaks=c(0.0075,0.015,0.032, 0.064,0.125,0.25, 0.5,1,2,4,8, 16))))


#to plot out later
uk.cip <- upset(myk.data.uk.mic.cip,
      cip.mech.list,
      sort_sets=FALSE,
      width_ratio=0.1,
      height_ratio=1,
      set_sizes=(
          upset_set_size(position='right')
          + theme(axis.text.x=element_text(angle=90))
          ),
      guides='over',
      base_annotations=list('Intersection size'=intersection_size(counts=FALSE,mapping=aes(fill=`CIP`)) + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26','I'='#fc9272'))),
      annotations = list(
          "MIC"=ggplot(mapping=aes(x=intersection, y=CIP_MIC_edit)) + ggtitle("UKHSA - Ciprofloxacin")  + geom_violin(aes(fill = `CIP`),width=1, alpha=1)  + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26','I'='#fc9272'), guide = 'none') +  scale_y_continuous(trans=log2_trans(), breaks=c(0.0075,0.015,0.032, 0.064,0.125,0.25, 0.5,1,2,4,8, 16))))

```


```{r SEAP upset CIP}

#id those with no data for CIP
myk.data.seap.dd %>% filter(`Ciprofloxacin DD` == "Not Available" | `Ciprofloxacin` == "Not performed") %>%  group_by(`Ciprofloxacin`,`Ciprofloxacin DD`) %>% summarise(n=n())

#filter those with data for CIP and make numeric
myk.data.seap.dd.cip <- myk.data.seap.dd %>% filter(`Ciprofloxacin DD` != "Not Available") %>% filter(`Ciprofloxacin` != "Not performed") %>% transform(`Ciprofloxacin DD` = as.numeric(`Ciprofloxacin DD`)) %>% separate(col = "ciprofloxacin", sep = ":", c("CIP","Cip Geno Mechanism"), remove = FALSE, fill = "right")

paste("Number of SEAP isolates with CIP Phenotype data = ", nrow(myk.data.seap.dd.cip), sep ="")

#jitter for individual points
upset(myk.data.seap.dd.cip,
      cip.mech.list,
      sort_sets=FALSE,
      width_ratio=0.1,
      set_sizes=(
          upset_set_size(position='right')
          + theme(axis.text.x=element_text(angle=90))
          ),
      guides='over',
      base_annotations=list('Intersection size'=intersection_size(counts=TRUE,mapping=aes(fill=`CIP`)) + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26','I'='#fc9272'))),
       annotations = list(
          "Disc Diffusion"=ggplot(mapping=aes(x=intersection, y=Ciprofloxacin.DD)) + ggtitle("CIP mechanisms v Disc Diff") + geom_jitter(alpha=0.2) + geom_violin(aes(fill = `CIP`),width=1, alpha=1)  + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26','I'='#fc9272'),guide = 'none')))


upset(myk.data.seap.dd.cip,
      cip.mech.list,
      sort_sets=FALSE,
      width_ratio=0.1,
      set_sizes=(
          upset_set_size(position='right')
          + theme(axis.text.x=element_text(angle=90))
          ),
      guides='over',
      base_annotations=list('Intersection size'=intersection_size(counts=TRUE,mapping=aes(fill=`CIP`)) + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26','I'='#fc9272'))),
       annotations = list(
          "Disc Diffusion"=ggplot(mapping=aes(x=intersection, y=Ciprofloxacin.DD), text = aes(size = 2)) + ggtitle("SEAP study - Ciprofloxacin")  + geom_violin(aes(fill = `CIP`),width=1, alpha=1)  + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26','I'='#fc9272'),guide = 'none')))



#to plot out later
seap.cip <-upset(myk.data.seap.dd.cip,
      cip.mech.list,
      sort_sets=FALSE,
      width_ratio=0.1,
      height_ratio=1,
      set_sizes=(
          upset_set_size(position='right')
          + theme(axis.text.x=element_text(angle=90))
          ),
      guides='over',
      base_annotations=list('Intersection size'=intersection_size(counts=FALSE,mapping=aes(fill=`CIP`)) + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26','I'='#fc9272'))),
       annotations = list(
          "Disc Diffusion"=ggplot(mapping=aes(x=intersection, y=Ciprofloxacin.DD), text = aes(size = 2)) + ggtitle("SEAP study - Ciprofloxacin")  + geom_violin(aes(fill = `CIP`),width=1, alpha=1)  + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26','I'='#fc9272'),guide = 'none')))




```

```{r CDC upset CIP}

#transform the CIP Rslt to numeric - needed for plotting - currently in character. 
myk.data.cdc.mic.cip <- myk.data.cdc.mic %>% transform(`CIP Rslt` = as.numeric(`CIP Rslt`))  %>% separate(col = "ciprofloxacin", sep = ":", c("CIP","Cip Geno Mechanism"), remove = FALSE, fill = "right")

#jitter for individual points
upset(myk.data.cdc.mic.cip,
      cip.mech.list,
      sort_sets=FALSE,
      width_ratio=0.1,
      set_sizes=(
          upset_set_size(position='right')
          + theme(axis.text.x=element_text(angle=90))
          ),
      guides='over',
      base_annotations=list('Intersection size'=intersection_size(counts=TRUE,mapping=aes(fill=`CIP`)) + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26','I'='#fc9272'))),
      annotations = list(
          "MIC"=ggplot(mapping=aes(x=intersection, y=`CIP.Rslt`)) + ggtitle("CDC - Ciprofloxacin") + geom_jitter(alpha=0.2)  + geom_violin(aes(fill = `CIP`),width=1, alpha=1)  + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26','I'='#fc9272'), guide = 'none') +  scale_y_continuous(trans=log2_trans(), breaks=c(0.0075,0.015,0.03, 0.06,0.12,0.25, 0.5,1,2,4,8, 16))))

upset(myk.data.cdc.mic.cip,
      cip.mech.list,
      sort_sets=FALSE,
      width_ratio=0.1,
      set_sizes=(
          upset_set_size(position='right')
          + theme(axis.text.x=element_text(angle=90))
          ),
      guides='over',
      base_annotations=list('Intersection size'=intersection_size(counts=TRUE,mapping=aes(fill=`CIP`)) + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26','I'='#fc9272'))),
      annotations = list(
          "MIC"=ggplot(mapping=aes(x=intersection, y=`CIP.Rslt`)) + ggtitle("CDC - Ciprofloxacin")  + geom_violin(aes(fill = `CIP`),width=1, alpha=1)  + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26','I'='#fc9272'), guide = 'none') +  scale_y_continuous(trans=log2_trans(), breaks=c(0.0075,0.015,0.03, 0.06,0.12,0.25, 0.5,1,2,4,8, 16))))

#to plot out later
cdc.cip <- upset(myk.data.cdc.mic.cip,
      cip.mech.list,
      sort_sets=FALSE,
      width_ratio=0.1,
      height_ratio=1,
      set_sizes=(
          upset_set_size(position='right')
          + theme(axis.text.x=element_text(angle=90))
          ),
      guides='over',
      base_annotations=list('Intersection size'=intersection_size(counts=FALSE,mapping=aes(fill=`CIP`)) + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26','I'='#fc9272'))),
      annotations = list(
          "MIC"=ggplot(mapping=aes(x=intersection, y=`CIP.Rslt`)) + ggtitle("CDC - Ciprofloxacin")  + geom_violin(aes(fill = `CIP`),width=1, alpha=1)  + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26','I'='#fc9272'), guide = 'none') +  scale_y_continuous(trans=log2_trans(), breaks=c(0.0075,0.015,0.03, 0.06,0.12,0.25, 0.5,1,2,4,8, 16))))
```



```{r CIP tables}

#table with MIC and overall gene profiles
#UKHSA
myk.data.uk.mic.cip %>% group_by(ciprofloxacin,CIP_MIC_edit) %>% summarise(n=n())
myk.data.uk.mic.cip %>% group_by(ciprofloxacin,CIP......R..0.06mg.L...Typhi) %>% summarise(n=n())

#SEAP
myk.data.seap.dd.cip %>% group_by(ciprofloxacin,Ciprofloxacin) %>% summarise(n=n())
myk.data.seap.dd.cip %>% group_by(ciprofloxacin,Ciprofloxacin.DD) %>% summarise(n=n())

#CDC
myk.data.cdc.mic %>% group_by(ciprofloxacin,`CIP Rslt`) %>% summarise(n=n())
myk.data.cdc.mic %>% group_by(ciprofloxacin,`CIP Concl`) %>% summarise(n=n())


df1 <- myk.data.uk.mic.cip %>% group_by(ciprofloxacin,CIP_MIC_edit) %>% summarise(n=n()) 
df2 <- myk.data.uk.mic.cip %>% group_by(ciprofloxacin,CIP......R..0.06mg.L...Typhi) %>% summarise(n=n())
df4<- myk.data.seap.dd.cip %>% group_by(ciprofloxacin,Ciprofloxacin) %>% summarise(n=n())
df3 <- myk.data.seap.dd.cip %>% group_by(ciprofloxacin,Ciprofloxacin.DD) %>% summarise(n=n())
df5 <- myk.data.cdc.mic %>% group_by(ciprofloxacin,`CIP Rslt`) %>% summarise(n=n())
df6<- myk.data.cdc.mic %>% group_by(ciprofloxacin,`CIP Concl`) %>% summarise(n=n())

#write out in case need
write_csv(df1, "Myk_UK_MICvalues_CIP_15012024.csv")
write_csv(df2, "Myk_UK_MICcall_CIP_15012024.csv")
write_csv(df3, "Myk_SEAP_DDvalues_CIP_15012024.csv")
write_csv(df4, "Myk_SEAP_DDcall_CIP_15012024.csv")
write_csv(df5, "Myk_CDC_MICvalues_CIP_15012024.csv")
write_csv(df6, "Myk_CDC_MICcall_CIP_15012024.csv")

```


```{r UK upset CEF}
#Other drug to focus on - Ceftriaxone

#the < and > for some of the numbers are problematics
#transform the CIP_MIC_edit to numeric - needed for plotting - currently in character. 
myk.data.uk.mic.cef <- myk.data.uk.mic %>%  mutate(CEF_MIC_edit =ifelse(`CRO - Ceftriaxone` == "<0.015", "0.0075",ifelse(`CRO - Ceftriaxone` == "<0.25", "0.125",ifelse(`CRO - Ceftriaxone` == ">64", "64",`CRO - Ceftriaxone`)))) %>% transform(CEF_MIC_edit = as.numeric(CEF_MIC_edit)) %>% separate(col = "ceftriaxone", sep = ":", c("CEF","CEF Geno Mechanism"), remove = FALSE, fill = "right")

#CEF list as - changes to .
cef.mech.list <- colnames(myk.data.uk.mic.cef)[c(67,63)]


#with fitter for individual points
upset(myk.data.uk.mic.cef,
      cef.mech.list,
      sort_sets=FALSE,
      width_ratio=0.1,
      set_sizes=(
          upset_set_size(position='right')
          + theme(axis.text.x=element_text(angle=90))
          ),
      guides='over',
      base_annotations=list('Intersection size'=intersection_size(counts=TRUE,mapping=aes(fill=`CEF`)) + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26'))),
      annotations = list(
          "MIC"=ggplot(mapping=aes(x=intersection, y=CEF_MIC_edit)) + ggtitle("UKHSA - Ceftriaxone") + geom_jitter(alpha=0.2) + geom_violin(aes(fill = `CEF`),width=1, alpha=1)  + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26'), guide = 'none') +  scale_y_continuous(trans=log2_trans(), breaks=c(0.0075,0.015,0.032, 0.064,0.125,0.25, 0.5,1,2,4,8, 16,32,64))))

upset(myk.data.uk.mic.cef,
      cef.mech.list,
      sort_sets=FALSE,
      width_ratio=0.1,
      set_sizes=(
          upset_set_size(position='right')
          + theme(axis.text.x=element_text(angle=90))
          ),
      guides='over',
      base_annotations=list('Intersection size'=intersection_size(counts=TRUE,mapping=aes(fill=`CEF`)) + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26'))),
      annotations = list(
          "MIC"=ggplot(mapping=aes(x=intersection, y=CEF_MIC_edit)) + ggtitle("UKHSA - Ceftriaxone") + geom_violin(aes(fill = `CEF`),width=1, alpha=1)  + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26'), guide = 'none') +  scale_y_continuous(trans=log2_trans(), breaks=c(0.0075,0.015,0.032, 0.064,0.125,0.25, 0.5,1,2,4,8, 16,32,64))))

uk.cef <- upset(myk.data.uk.mic.cef,
      cef.mech.list,
      sort_sets=FALSE,
      width_ratio=0.1,
      height_ratio=1,
      set_sizes=(
          upset_set_size(position='right')
          + theme(axis.text.x=element_text(angle=90))
          ),
      guides='over',
      base_annotations=list('Intersection size'=intersection_size(counts=FALSE,mapping=aes(fill=`CEF`)) + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26'))),
      annotations = list(
          "MIC"=ggplot(mapping=aes(x=intersection, y=CEF_MIC_edit)) + ggtitle("UKHSA - Ceftriaxone") + geom_violin(aes(fill = `CEF`),width=1, alpha=1)  + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26'), guide = 'none') +  scale_y_continuous(trans=log2_trans(), breaks=c(0.0075,0.015,0.032, 0.064,0.125,0.25, 0.5,1,2,4,8, 16,32,64))))

```

```{r SEAP CEF}

#id those with no data for CEF
myk.data.seap.dd %>% filter(`Ceftriaxone DD` == "Not Available" | `Ceftriaxone` == "Not performed") %>%  group_by(`Ceftriaxone`,`Ceftriaxone DD`) %>% summarise(n=n())


#filter those with data for CEF and make numeric
myk.data.seap.dd.cef <- myk.data.seap.dd %>% filter(`Ceftriaxone DD` != "Not Available") %>% filter(`Ceftriaxone` != "Not performed") %>% transform(`Ceftriaxone DD` = as.numeric(`Ceftriaxone DD`)) %>% separate(col = "ceftriaxone", sep = ":", c("CEF","CEF Geno Mechanism"), remove = FALSE, fill = "right")

paste("Number of SEAP isolates with CEF Phenotype data = ", nrow(myk.data.seap.dd.cef), sep ="")


#jitter for individual points
upset(myk.data.seap.dd.cef,
      cef.mech.list,
      sort_sets=FALSE,
      width_ratio=0.1,
      set_sizes=(
          upset_set_size(position='right')
          + theme(axis.text.x=element_text(angle=90))
          ),
      guides='over',
      base_annotations=list('Intersection size'=intersection_size(counts=TRUE,mapping=aes(fill=`CEF`)) + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26'))),
       annotations = list(
          "Disc Diffusion"=ggplot(mapping=aes(x=intersection, y=Ceftriaxone.DD)) + ggtitle("CEF mechanisms v Disc Diff") + geom_jitter(alpha=0.2) + geom_violin(aes(fill = `CEF`),width=1, alpha=1)  + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26'),guide = 'none')))

#jitter for individual points
upset(myk.data.seap.dd.cef,
      cef.mech.list,
      sort_sets=FALSE,
      width_ratio=0.1,
      set_sizes=(
          upset_set_size(position='right')
          + theme(axis.text.x=element_text(angle=90))
          ),
      guides='over',
      base_annotations=list('Intersection size'=intersection_size(counts=TRUE,mapping=aes(fill=`CEF`)) + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26'))),
       annotations = list(
          "Disc Diffusion"=ggplot(mapping=aes(x=intersection, y=Ceftriaxone.DD)) + ggtitle("SEAP study - Ceftriaxone") + geom_violin(aes(fill = `CEF`),width=1, alpha=1)  + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26'),guide = 'none')))


#to plot out later
seap.cef <-upset(myk.data.seap.dd.cef,
      cef.mech.list,
      sort_sets=FALSE,
      width_ratio=0.1,
      height_ratio=1,
      set_sizes=(
          upset_set_size(position='right')
          + theme(axis.text.x=element_text(angle=90))
          ),
      guides='over',
      base_annotations=list('Intersection size'=intersection_size(counts=FALSE,mapping=aes(fill=`CEF`)) + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26'))),
       annotations = list(
          "Disc Diffusion"=ggplot(mapping=aes(x=intersection, y=Ceftriaxone.DD)) + ggtitle("SEAP study - Ceftriaxone") + geom_violin(aes(fill = `CEF`),width=1, alpha=1)  + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26'),guide = 'none')))



```

```{r CDC upset CEF}
#AXO	Ceftriaxone
#transform the AXO Rslt to numeric - needed for plotting - currently in character. 
myk.data.cdc.mic.cef <- myk.data.cdc.mic %>% transform(`AXO Rslt` = as.numeric(`AXO Rslt`))  %>% separate(col = "ceftriaxone", sep = ":", c("CEF","Cef Geno Mechanism"), remove = FALSE, fill = "right")

#jitter for individual points
upset(myk.data.cdc.mic.cef,
      cef.mech.list,
      sort_sets=FALSE,
      width_ratio=0.1,
      set_sizes=(
          upset_set_size(position='right')
          + theme(axis.text.x=element_text(angle=90))
          ),
      guides='over',
      base_annotations=list('Intersection size'=intersection_size(counts=TRUE,mapping=aes(fill=`CEF`)) + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26'))),
      annotations = list(
          "MIC"=ggplot(mapping=aes(x=intersection, y=`AXO.Rslt`)) + ggtitle("CDC - Ceftriaxone") + geom_jitter(alpha=0.2)  + geom_violin(aes(fill = `CEF`),width=1, alpha=1)  + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26'), guide = 'none') +  scale_y_continuous(trans=log2_trans(), breaks=c(0.0075,0.015,0.03, 0.06,0.12,0.25, 0.5,1,2,4,8, 16,32,64))))

upset(myk.data.cdc.mic.cef,
      cef.mech.list,
      sort_sets=FALSE,
      width_ratio=0.1,
      set_sizes=(
          upset_set_size(position='right')
          + theme(axis.text.x=element_text(angle=90))
          ),
      guides='over',
      base_annotations=list('Intersection size'=intersection_size(counts=TRUE,mapping=aes(fill=`CEF`)) + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26'))),
      annotations = list(
          "MIC"=ggplot(mapping=aes(x=intersection, y=`AXO.Rslt`)) + ggtitle("CDC - Ceftriaxone")  + geom_violin(aes(fill = `CEF`),width=1, alpha=1)  + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26'), guide = 'none') +  scale_y_continuous(trans=log2_trans(), breaks=c(0.0075,0.015,0.03, 0.06,0.12,0.25, 0.5,1,2,4,8, 16,32,64))))

#to plot out later
cdc.cef <- upset(myk.data.cdc.mic.cef,
      cef.mech.list,
      sort_sets=FALSE,
      width_ratio=0.1,
      height_ratio=1,
      set_sizes=(
          upset_set_size(position='right')
          + theme(axis.text.x=element_text(angle=90))
          ),
      guides='over',
      base_annotations=list('Intersection size'=intersection_size(counts=FALSE,mapping=aes(fill=`CEF`)) + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26'))),
      annotations = list(
          "MIC"=ggplot(mapping=aes(x=intersection, y=`AXO.Rslt`)) + ggtitle("CDC - Ceftriaxone")  + geom_violin(aes(fill = `CEF`),width=1, alpha=1)  + scale_fill_manual(values=c('S'='#fee0d2','R'='#de2d26'), guide = 'none') +  scale_y_continuous(trans=log2_trans(), breaks=c(0.0075,0.015,0.03, 0.06,0.12,0.25, 0.5,1,2,4,8, 16,32,64))))
```


```{r CEF tables}

#table with MIC and overall gene profiles
myk.data.uk.mic %>% group_by(ceftriaxone,`CRO - Ceftriaxone`) %>% summarise(n=n())
myk.data.uk.mic %>% group_by(ceftriaxone,`CRO  -    R> 0.5mg/L`) %>% summarise(n=n())
myk.data.seap.dd %>% group_by(ceftriaxone,Ceftriaxone) %>% summarise(n=n())
myk.data.seap.dd %>% group_by(ceftriaxone,`Ceftriaxone DD`,Ceftriaxone) %>% summarise(n=n())
myk.data.cdc.mic %>% group_by(ceftriaxone,`AXO Rslt`) %>% summarise(n=n())
myk.data.cdc.mic %>% group_by(ceftriaxone,`AXO Concl`) %>% summarise(n=n())


df7 <- myk.data.uk.mic %>% group_by(ceftriaxone,`CRO - Ceftriaxone`) %>% summarise(n=n())
df8 <- myk.data.uk.mic %>% group_by(ceftriaxone,`CRO  -    R> 0.5mg/L`) %>% summarise(n=n())
df9 <- myk.data.seap.dd %>% group_by(ceftriaxone,`Ceftriaxone DD`,Ceftriaxone) %>% summarise(n=n())
df10 <- myk.data.seap.dd %>% group_by(ceftriaxone,Ceftriaxone) %>% summarise(n=n())
df11 <- myk.data.cdc.mic %>% group_by(ceftriaxone,`AXO Concl`) %>% summarise(n=n())
df12 <- myk.data.cdc.mic %>% group_by(ceftriaxone,`AXO Rslt`) %>% summarise(n=n())

#write out in case need
write_csv(df7, "Myk_UK_MICvalues_CEF_15012024.csv")
write_csv(df8, "Myk_UK_MICcall_CEF_15012024.csv")
write_csv(df9, "Myk_SEAP_DDvalues_CEF_15012024.csv")
write_csv(df10, "Myk_SEAP_DDcall_CEF_15012024.csv")
write_csv(df11, "Myk_CDC_MICvalues_CEF_15012024.csv")
write_csv(df12, "Myk_CDC_MICcall_CEF_15012024.csv")

```


```{r numbers for the different drugs across studies }
#summary of Mykrobe AMR profile + reported phenotypic resistance across the three studies

#Azithromycin
#UKHSA - one isolate - susceptible pheno
myk.data.uk.mic %>% group_by(azithromycin,`AZM - Azithromycin`) %>% summarise(n=n())
myk.data.uk.mic%>% group_by(azithromycin,`AZM-    R> 16mg/L`) %>% summarise(n=n())
#SEAP - number discordant
myk.data.seap.dd %>% group_by(azithromycin,Azithromycin) %>% summarise(n=n())
myk.data.seap.dd %>% group_by(azithromycin,`Azithromycin DD`,Azithromycin) %>% summarise(n=n())
#CDC - two matched - check MIC breakpoint - same as UKHSA
myk.data.cdc.mic %>% group_by(azithromycin,`AZM Rslt`) %>% summarise(n=n())
myk.data.cdc.mic %>% group_by(azithromycin,`AZM Concl`) %>% summarise(n=n())

#Chloramphenicol
#UKHSA - all matched
myk.data.uk.mic %>% group_by(chloramphenicol,`CHL - Chloramphenicol`) %>% summarise(n=n())
myk.data.uk.mic%>% group_by(chloramphenicol,`CHL-    R> 8mg/L`) %>% summarise(n=n())
#SEAP - some discordant ones
myk.data.seap.dd %>% group_by(chloramphenicol,Chloramphenicol) %>% summarise(n=n())
myk.data.seap.dd %>% group_by(chloramphenicol,`Chloramphenicol DD`,Chloramphenicol) %>% summarise(n=n())
#CDC - all matched
myk.data.cdc.mic %>% group_by(chloramphenicol,`CHL Rslt`) %>% summarise(n=n())
myk.data.cdc.mic %>% group_by(chloramphenicol,`CHL Concl`) %>% summarise(n=n())

#Ampicillin
#UKHSA - 5 discrepencies - R pheno but no mechanism detected
myk.data.uk.mic %>% group_by(ampicillin,`AMX - Amoxicillin`) %>% summarise(n=n())
myk.data.uk.mic %>% group_by(ampicillin,`AMX -   R> 8mg/L`,`AMX - Amoxicillin`) %>% summarise(n=n())
#SEAP - some mismatch
myk.data.seap.dd %>% group_by(ampicillin,`Ampicillin`) %>% summarise(n=n())
myk.data.seap.dd %>% group_by(ampicillin,`Ampicillin DD`,Ampicillin) %>% summarise(n=n())
#CDC - two discordant
myk.data.cdc.mic %>% group_by(ampicillin, `AMP Rslt`) %>% summarise(n=n())
myk.data.cdc.mic %>% group_by(ampicillin,`AMP Concl`) %>% summarise(n=n())

#Tetracycline - only CDC and UKHSA - no discrepencies
#UKHSA - all match
myk.data.uk.mic %>% group_by(tetracycline,`TET - Tetracycline`) %>% summarise(n=n())
myk.data.uk.mic %>% group_by(tetracycline,`TET-    R> 32mg/L`) %>% summarise(n=n())
#SEAP - not tested for tetracycline
#CDC - all match
myk.data.cdc.mic %>% group_by(tetracycline,`TET Rslt`) %>% summarise(n=n())
myk.data.cdc.mic %>% group_by(tetracycline,`TET Concl`) %>% summarise(n=n())


#Sulfamethoxazole
#UKHSA - five isolates discordant
myk.data.uk.mic %>% group_by(sulfamethoxazole,`SXT - Trimethoprim-Sulfamethoxazole`) %>% summarise(n=n())
myk.data.uk.mic%>% group_by(sulfamethoxazole,`SXT -  S ≤ 2,  R > 4 mg/L`) %>% summarise(n=n())
#SEAP - multiple discordant
myk.data.seap.dd %>% group_by(sulfamethoxazole,Cotrimoxazole) %>% summarise(n=n())
myk.data.seap.dd %>% group_by(sulfamethoxazole,`Cotrimoxazole DD`,Cotrimoxazole) %>% summarise(n=n())
#CDC - 1 discordant isolate
myk.data.cdc.mic %>% group_by(sulfamethoxazole,`COT Rslt`) %>% summarise(n=n())
myk.data.cdc.mic %>% group_by(sulfamethoxazole,`COT Concl`) %>% summarise(n=n())


```


```{r upset plots }

#CIP upset plot
uk.cip | seap.cip | cdc.cip

#CEF upset plot
uk.cef | seap.cef | cdc.cef

#pdf
pdf('Upset_plots_Ciprofloxacin_15012024.pdf', h =12, w = 14)
uk.cip | seap.cip | cdc.cip
dev.off()

pdf('Upset_plots__Ceftriaxone_15012024.pdf', h =10, w = 10)
uk.cef | seap.cef | cdc.cef
dev.off()

```
```