---
title: "Mykrobe_ONT_2024"
author: "DanielleIngle"
date: "09/01/2024 - 11/01/2024"
output: html_document
---
---
```{r setup, include=FALSE}
require('knitr')
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = getwd())
getwd()

```

```{r library}

library(tidyverse)
library(readxl)
library(ggplot2)
library(patchwork)
library(RColorBrewer)

sessionInfo()
```

```{r files}
#metadata file with info for AMR 
meta.tgc <- read_csv("/Users/daniellei/Documents/github/TyphoidGenomicsConsortiumWG2/TGC_data.csv")

#ONT details 
meta.ont <- read_excel("../ONTanalysis_july2023/ONTdata_processed_details_10012024.xlsx")

#ONT results file from Mykrobe 
ont <- read_tsv('../ONTanalysis_july2023/Mykrobe_ONTtyphi_23082023_predictResults.tsv')

#Illumina data results file from Mykrobe
ill <- read_tsv('../ONTanalysis_july2023/Mykrobe_shortread_Typhi_10012024_n96_predictResults.tsv')

```

```{r merge the four files together}
#subset and then append colnames on TGC data so know it was the TGC genotype
meta.tgc.sub <- meta.tgc[,c(1:7,9:10,12,103)]
colnames(meta.tgc.sub) <- paste("TGC", colnames(meta.tgc.sub), sep = "_")

meta.ont.tgc <- meta.ont %>% left_join(meta.tgc.sub, by =c("TGC_sangerlane" = "TGC_TGC_sangerlane"))

#add in the ONT or Illumina to mykrobe files - so know where came from. 
#the short read one - only for the matched isolates with no report genotype - new isolates. Otherwise use those reported in the WG1 eLIFE paper. 

ont$Seq_type <- "ONT"
ill$Seq_type <- "Illumina"

#append colnames on ont data so know it was the ont results
colnames(ont) <- paste("ONT", colnames(ont), sep = "_")

#add in the ONT mykrobe results 
ont.data <- meta.ont.tgc %>% left_join(ont, by = c('MykrobeRunNamesLONG' = 'ONT_genome') )

#add in the short read data results
#ifelse - if isolates in TGC spreadsheet - use that. Else add in the new mykrobe calls. 
table(ont.data$GenotypeShortRequired)
#expected genotype - pulled from spreadsheet originally - merge in here as cleaner
table(ont.data$Expected_Genotype, useNA = "ifany")

ont.ill.data.total <- ont.data %>% left_join(ill, by = c('Illumina reads accession' = 'genome') )

write_csv(ont.ill.data.total, "ONT_Illumina_TyphiMykrobe_Calls_11102024.csv")

```
```{r filter out those where lack data for both}

ont.ill.data.total %>% group_by(`Data both read types`) %>% summarise(n=n())
ont.ill.data <- ont.ill.data.total %>% filter(`Data both read types` == "Y")
ont.ill.data.notboth <- ont.ill.data.total %>% filter(`Data both read types` == "N")

paste("number of isolates with both matched long and short read data = ", nrow(ont.ill.data), sep = "")
paste("number of isolates with missing either long or short read data = ", nrow(ont.ill.data.notboth), sep = "")

write_csv(ont.ill.data.notboth, "ONT_Illumina_TyphiMykrobe_NotBothReadTypes_11102024.csv")

```


```{r genotype matching}

#filter to just the different genotypes and confidence - so can check these are good / why there are problems
geno.sub <- ont.ill.data[,c(1:12,35:44,110:118)]

geno.sub$genoMatch <- ifelse(geno.sub$ONT_genotype == geno.sub$genotype, geno.sub$genotype, "NoMatch") 
geno.sub$SppMatch <- ifelse(geno.sub$ONT_species == geno.sub$species, geno.sub$species, "NoMatch") 
geno.sub %>% group_by(SppMatch) %>% summarise(n=n())
geno.sub %>% group_by(genoMatch) %>% summarise(n=n())

#filtering out those without Typhi species match
geno.sub %>% filter(SppMatch == "typhi") %>% group_by(SppMatch) %>% summarise(n=n())
geno.sub %>% filter(SppMatch == "typhi") %>% group_by(genoMatch) %>% summarise(n=n())

geno.match.N <- geno.sub %>% filter(SppMatch == "typhi") %>% group_by(genoMatch) %>% summarise(n=n())
geno.NOTmatch <- geno.sub %>% filter(SppMatch == "typhi") %>% filter(genoMatch == "NoMatch")

write_csv(geno.sub, "ONT_Illumina_TyphiMykrobe_compareCalls_11012024.csv")
write_csv(geno.NOTmatch, "ONT_Illumina_TyphiMykrobe_compareCalls_NotMatch_n3_11012024.csv")
write_csv(geno.match.N, "ONT_Illumina_TyphiMykrobe_GenotypesIncluded_11012024.csv")

```


```{r filter typhi isolates}

#filter out the isolate that didn't have a species Typhi call - can't compare it - limitation of possibly not extracted exact same sample
ont.ill.data %>% group_by(ONT_species,species) %>% summarise(n=n())

 
ont.ill.data.typhi <- ont.ill.data %>% filter(ONT_species == "typhi") %>% filter(species == "typhi")
nrow(ont.ill.data.typhi)
ont.ill.data.NOTtyphi <- ont.ill.data %>% filter(ONT_species != "typhi" | species != "typhi")


paste("number of isolates with Typhi calls for both long and short = ", nrow(ont.ill.data.typhi), sep = "")
paste("number of isolates with 'not Typhi' for either long or short read data = ", nrow(ont.ill.data.NOTtyphi), sep = "")

write_csv(ont.ill.data.NOTtyphi, "NotTyphiCalls_ONT_Illumina_TyphiMykrobe_n6_11012024.csv")

```

```{r AMR matching - class}

#Looking at the AMR
#Ceftriaxone
ont.ill.data.typhi$ceftriaxone_call <- ifelse(ont.ill.data.typhi$ONT_ceftriaxone == ont.ill.data.typhi$ceftriaxone, ont.ill.data.typhi$ceftriaxone, "NoMatch") 
ont.ill.data.typhi %>% group_by(ceftriaxone_call) %>% summarise(n=n())

#Azithromycin
ont.ill.data.typhi$azithromycin_call <- ifelse(ont.ill.data.typhi$ONT_azithromycin == ont.ill.data.typhi$azithromycin, ont.ill.data.typhi$azithromycin, "NoMatch") 
ont.ill.data.typhi %>% group_by(azithromycin_call) %>% summarise(n=n())

#Ciprofloxacin
ont.ill.data.typhi$ciprofloxacin_call <- ifelse(ont.ill.data.typhi$ONT_ciprofloxacin == ont.ill.data.typhi$ciprofloxacin, ont.ill.data.typhi$ciprofloxacin, "NoMatch") 
ont.ill.data.typhi %>% group_by(ciprofloxacin_call) %>% summarise(n=n())
#has n = 8 no Match - look at these ones

#ampicillin
ont.ill.data.typhi$ampicillin_call <- ifelse(ont.ill.data.typhi$ONT_ampicillin == ont.ill.data.typhi$ampicillin, ont.ill.data.typhi$ampicillin, "NoMatch") 
ont.ill.data.typhi %>% group_by(ampicillin_call) %>% summarise(n=n())

#chloramphenicol
ont.ill.data.typhi$chloramphenicol_call <- ifelse(ont.ill.data.typhi$ONT_chloramphenicol == ont.ill.data.typhi$chloramphenicol, ont.ill.data.typhi$chloramphenicol, "NoMatch") 
ont.ill.data.typhi %>% group_by(chloramphenicol_call) %>% summarise(n=n())
#has n = 1 no Match - look at these ones

#tetracycline
ont.ill.data.typhi$tetracycline_call <- ifelse(ont.ill.data.typhi$ONT_tetracycline == ont.ill.data.typhi$tetracycline, ont.ill.data.typhi$tetracycline, "NoMatch") 
ont.ill.data.typhi %>% group_by(tetracycline_call) %>% summarise(n=n())

#co-trim
ont.ill.data.typhi$CoTrim_call <- ifelse(ont.ill.data.typhi$`ONT_trimethoprim-sulfamethoxazole` == ont.ill.data.typhi$`trimethoprim-sulfamethoxazole`, ont.ill.data.typhi$`trimethoprim-sulfamethoxazole`, "NoMatch") 
ont.ill.data.typhi %>% group_by(CoTrim_call) %>% summarise(n=n())


```

```{r plasmid matching - specific}

#check which plasmid reps have something detected in either short or long read data. 

ont.ill.data.typhi %>% group_by(ONT_IncFIAHI1,IncFIAHI1) %>% summarise(n=n())
ont.ill.data.typhi %>% group_by(ONT_IncHI1A,IncHI1A) %>% summarise(n=n())
ont.ill.data.typhi %>% group_by(ONT_IncHI1BR27,IncHI1BR27) %>% summarise(n=n())
ont.ill.data.typhi %>% group_by(ONT_IncHI1_ST6,IncHI1_ST6) %>% summarise(n=n())
ont.ill.data.typhi %>% group_by(ONT_IncY,IncY) %>% summarise(n=n())
ont.ill.data.typhi %>% group_by(ONT_IncX3,IncX3) %>% summarise(n=n())
ont.ill.data.typhi %>% group_by(ONT_IncHI2A,IncHI2A) %>% summarise(n=n())
ont.ill.data.typhi %>% group_by(ONT_IncI1,IncI1) %>% summarise(n=n())
ont.ill.data.typhi %>% group_by(ONT_IncL_M,IncL_M) %>% summarise(n=n())
ont.ill.data.typhi %>% group_by(ONT_IncFIB_pHCM2,IncFIB_pHCM2) %>% summarise(n=n())
ont.ill.data.typhi %>% group_by(ONT_IncFIB_K,IncFIB_K) %>% summarise(n=n())
ont.ill.data.typhi %>% group_by(ONT_IncN,IncN) %>% summarise(n=n())
ont.ill.data.typhi %>% group_by(ONT_z66,z66) %>% summarise(n=n())

#plasmid mismatch 
#IncHI1A - n = 1
#IncFIB_pHCM2 - n = 3

#IncHI1A_call
ont.ill.data.typhi$IncHI1A_call <- ifelse(ont.ill.data.typhi$`ONT_IncHI1A` == ont.ill.data.typhi$IncHI1A, ont.ill.data.typhi$IncHI1A, "NoMatch") 
ont.ill.data.typhi %>% group_by(IncHI1A_call) %>% summarise(n=n())

#IncFIB_pHCM2 call 
ont.ill.data.typhi$IncFIB_pHCM2_call <- ifelse(ont.ill.data.typhi$`ONT_IncFIB_pHCM2` == ont.ill.data.typhi$IncFIB_pHCM2, ont.ill.data.typhi$IncFIB_pHCM2, "NoMatch") 
ont.ill.data.typhi %>% group_by(IncFIB_pHCM2_call) %>% summarise(n=n())


```

```{r discordant calls}


#those with AMR and plasmid discordant calls - filer across the Matching cols for all drugs and plasmid reps
discord <- ont.ill.data.typhi %>% filter(if_any(everything(), ~ .x == "NoMatch"))
discord.class <- discord[,c(1:12,184:192, 48:49,122:123,55,129,63,137)]


#write out
write_csv(ont.ill.data.typhi, "ONT_Illumina_TyphiMykrobe_compareCalls_AMRplasmid_11012024.csv")
write_csv(discord, "ONT_Illumina_TyphiMykrobe_discordAMRplasmid_11012024.csv")
write_csv(discord.class, "ONT_Illumina_TyphiMykrobe_discordAMRplasmid__Subset_n12_11012024.csv")

```


```{r numbers}
#numbers for sequence data
paste("Original number of isolates in the dataset = ", nrow(ont.ill.data.total), sep = "")
paste("Number of isolates with matched data of ONT and short = ", nrow(ont.ill.data), sep = "")
paste("Number of isolates where need either ONT or short data = ", nrow(ont.ill.data.notboth), sep = "")

#using on the n = 90 isolates 
#species - both calls typhi
paste("Number of isolates with species match = ", nrow(ont.ill.data.typhi), sep = "")
paste("Number of isolates that don't species match  = ", nrow(ont.ill.data.NOTtyphi), sep = "")
paste("Percentage of isolates called as Typhi with both ONT and short read data  = ", nrow(ont.ill.data.typhi)/nrow(ont.ill.data)*100, sep = "")

#number of isolates with genotype match - of the 84 with species calls. 
paste("Number of isolates with genotype match = ", sum(geno.match.N$n)-nrow(geno.NOTmatch) , sep = "")
paste("Number of isolates that don't genotype match  = ", nrow(geno.NOTmatch), sep = "")
paste("Percentage of isolates with matched genotype calls from both ONT and short read data  = ", (sum(geno.match.N$n)-nrow(geno.NOTmatch))/nrow(ont.ill.data.typhi)*100, sep = "")

#AMR numbers - from the n =84. The ones that mismatch on genotype - didn't have and differences on AMR and plasmid

paste("Number of isolates with AMR and plasmid match = ", nrow(ont.ill.data.typhi)-nrow(discord) , sep = "")
paste("Number of isolates that have either AMR or plasmid mismatch  = ", nrow(discord), sep = "")
paste("Percentage of isolates with matched AMR and plasmid calls from both ONT and short read data  = ", (sum(geno.match.N$n)-nrow(geno.NOTmatch))/nrow(ont.ill.data.typhi)*100, sep = "")

# 3 plasmid mismatch.
#1 isolate with loss incHIA and catA gene
# three isolates with loss of IncFIB_pHCM2 
# 8 isolates with Cip point mutations mismatch. 1 India, 1 Australia and 6 Malawi


```


```{r plots}

plot.cols <- brewer.pal(name = "Paired", n = 10)

species.plot <- ont.ill.data %>% group_by(ONT_species,species) %>% summarise(n=n())

ggplot(species.plot, aes(x=`ONT_species`, y=n,fill = `species`)) + geom_bar(stat="identity")  + theme(axis.title = element_text(face="bold", colour="black", size=14), axis.text.x  = element_text(angle=45, vjust=0.5,size=12),axis.text.y  = element_text(size=12))     + theme(legend.position="none")  + scale_fill_manual(values = plot.cols[5:6]) + ggtitle("ONT species call \nn = 90 isolates") 

p1 <- ggplot(species.plot, aes(x=`ONT_species`, y=n,fill = `species`)) + geom_bar(stat="identity")  + theme(axis.title = element_text(face="bold", colour="black", size=14), axis.text.x  = element_text(angle=45, vjust=0.5,size=12),axis.text.y  = element_text(size=12))     + theme(legend.position="none") + scale_fill_manual(values = plot.cols[5:6]) + ggtitle("ONT species call \nn = 90 isolates") + ylim(0, 100)


#AMR, genotype and plasmid 
gt <- data.frame(
  Concordance= factor(c("Match1","Not Match1"), levels=c("Match1","Not Match1")),
  n = c(81,3), Feature = c("Genotype", "Genotype")
)

amr <- data.frame(
  Concordance= factor(c("Match2","Not Match2"), levels=c("Match2","Not Match2")),
  n = c(75,9), Feature = c("AMR", "AMR")
)


plas <- data.frame(
  Concordance= factor(c("Match3","Not Match3"), levels=c("Match3","Not Match3")),
  n = c(80,4), Feature = c("Plasmid", "Plasmid")
)


concord.plot <-rbind(gt, amr,plas)
concord.plot$Feature <- factor(concord.plot$Feature, levels=c("Genotype","AMR","Plasmid" ))

ggplot(concord.plot, aes(x=`Feature`, y=n,fill = `Concordance`)) + geom_bar(stat="identity", position=position_dodge())  + theme(axis.title = element_text(face="bold", colour="black", size=14), axis.text.x  = element_text(angle=45, vjust=0.5,size=12),axis.text.y  = element_text(size=12))     + theme(legend.position="none") + scale_fill_manual(values = rev(plot.cols[c(1:4,9:10)])) + ggtitle("ONT concordance with Illumina calls \nn = 84 isolates") 

p2 <- ggplot(concord.plot, aes(x=`Feature`, y=n,fill = `Concordance`)) + geom_bar(stat="identity", position=position_dodge())  + theme(axis.title = element_text(face="bold", colour="black", size=14), axis.text.x  = element_text(angle=45, vjust=0.5,size=12),axis.text.y  = element_text(size=12))     + theme(legend.position="none") + scale_fill_manual(values = rev(plot.cols[c(1:4,9:10)])) + ggtitle("ONT concordance with Illumina calls \nn = 84 isolates") + ylim(0, 100)


p1 + p2 +  plot_layout(widths = c(1, 3))

pdf("ONT_concordance_18012024.pdf")
p1 + p2 +  plot_layout(widths = c(1, 3))
dev.off()
```