---
title: "Global Typhoid Genomics Consortium - Mykrobe - AMR and plasmid typing validation"
author: "Danielle Ingle, Kat Holt"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
  prettydoc::html_pretty:
    theme: cayman
    toc: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# set working directory to the location of the script
knitr::opts_knit$set(root.dir = getwd())
setwd(dirname(rstudioapi::getSourceEditorContext()$path))

# load packages
library(knitr)
library(tidyverse)
library(dplyr)
library(ggplot2)
library(patchwork)
library(ggsankey)
```

# print session info (package versions)
```{r session_info}

# print versions
sessionInfo()

```

# read genotype calls from Mykrobe and PW
``` {r read_data}
# read as a tibble
myk_data <- read_csv("../Supplementary_Table_1.csv")
valid_data <- read_csv("../Supplementary_Table_2.csv")

colnames(myk_data)[2:75] <- paste("MYK", colnames(myk_data)[2:75], sep = "_")
amr <- myk_data %>% left_join(valid_data, by = "TGC_ID")

#filter to where have calls for both Mykrobe and PW 
amr <- amr %>% filter(MYK_species == "typhi")  %>% filter(`PW_Species Name` == "Salmonella enterica")
```

# compare ESBL calls
``` {r esbl}
# which ESBLs did Mykrobe call? CTX-M-15, SHV-12
amr %>% select(c(`MYK_blaCTX-M-15`, `MYK_blaSHV-12`, `MYK_blaOXA-134`, `MYK_AmpC1`)) %>% group_by(pick(everything())) %>% count()

amr %>% select(c(`MYK_blaCTX-M-15`,starts_with("PW_blaCTX")))  %>% group_by(pick(everything())) %>% count()

amr %>% select(c(`MYK_blaSHV-12`,starts_with("PW_blaSHV")))  %>% group_by(pick(everything())) %>% count()

# OXAs not detected by Mykrobe, but 1 (ampR, not ESBL) detected by PW
amr %>% select(c(starts_with("MYK_blaOXA"),starts_with("PW_blaOXA")))  %>% group_by(pick(everything())) %>% count()

# no AmpC detected by either
amr %>% select(c(`MYK_AmpC1`,starts_with("PW_amp")))  %>% group_by(pick(everything())) %>% count()

```

# compare ampicillin calls
``` {r amp}
# check TEM calls
amr %>% select(`MYK_blaTEM-1D`, `PW_blaTEM-1D` ) %>% group_by(pick(everything())) %>% count()

# OXAs not detected by Mykrobe, but 1 (ampR, not ESBL) detected by PW
amr %>% select(c(starts_with("MYK_blaOXA"),starts_with("PW_blaOXA")))  %>% group_by(pick(everything())) %>% count()

```

# compare azi calls
``` {r azi}
# check acrB calls
amr %>% select(starts_with("MYK_acr"), starts_with("PW_acrB")) %>% group_by(pick(everything())) %>% count()

```

# compare chloramph calls
``` {r chl}
# check cat calls - one cmlA1, agreed by both methods, in a catA1+ strain
amr %>% select(MYK_catA1,MYK_cmlA1, PW_catA1, PW_cmlA) %>% group_by(pick(everything())) %>% count()

amr %>% select(MYK_catA1, PW_catA1) %>% group_by(pick(everything())) %>% count()

amr %>% select(MYK_cmlA1, PW_cmlA) %>% group_by(pick(everything())) %>% count()
```

# compare cip calls
``` {r cip}
amr %>% select(starts_with("MYK_qnr"), starts_with("MYK_gyr"), starts_with("MYK_par")) %>% colSums()

amr %>% select(starts_with("MYK_qnr"), starts_with("PW_qnr")) %>% group_by(pick(everything())) %>% count()

amr %>% select(starts_with("MYK_qnrS1"), starts_with("PW_qnrS")) %>% group_by(pick(everything())) %>% count()

amr %>% select(starts_with("MYK_qnrB1"), starts_with("PW_qnrB")) %>% group_by(pick(everything())) %>% count()
  
amr %>% select(starts_with("MYK_qnrD1"), starts_with("PW_qnrD")) %>% group_by(pick(everything())) %>% count()

amr %>% select(starts_with("MYK_gyrA_S83"), starts_with("PW_gyrA_S83")) %>% group_by(pick(everything())) %>% count()

amr %>% select(starts_with("MYK_gyrA_D87"), starts_with("PW_gyrA_D87")) %>% group_by(pick(everything())) %>% count()

amr %>% select(starts_with("MYK_gyrA_D87Y"), starts_with("PW_gyrA_D87")) %>% group_by(pick(everything())) %>% count()

amr %>% select(starts_with("MYK_gyrA_D87G"), starts_with("PW_gyrA_D87")) %>% group_by(pick(everything())) %>% count()

amr %>% select(starts_with("MYK_gyrA_D87V"), starts_with("PW_gyrA_D87")) %>% group_by(pick(everything())) %>% count()

amr %>% select(starts_with("MYK_gyrA_D87N"), starts_with("PW_gyrA_D87")) %>% group_by(pick(everything())) %>% count()

amr %>% select(starts_with("MYK_parC_S80"), starts_with("PW_parC_S80")) %>% group_by(pick(everything())) %>% count()

amr %>% select(starts_with("MYK_parC_E84"), starts_with("PW_parC_E84")) %>% group_by(pick(everything())) %>% count()

amr %>% select(starts_with("MYK_gyrB"), starts_with("PW_gyrB")) %>% group_by(pick(everything())) %>% count()

```


# compare sul calls
``` {r sul}
# check sul calls 
amr %>% select(MYK_sul1, MYK_sul2, PW_sul1, PW_sul2) %>% colSums()

amr %>% select(MYK_sul1, PW_sul1) %>% group_by(pick(everything())) %>% count()

amr %>% select(MYK_sul2, PW_sul2) %>% group_by(pick(everything())) %>% count()

```


# compare dfr calls
``` {r dfr}
# check dfr calls 
amr %>% select(starts_with("MYK_dfr")) %>% colSums()

amr %>% select(starts_with("MYK_dfr")) %>% group_by(pick(everything()))

dfr_hits_MYK <- amr %>% select(starts_with("MYK_dfr")) %>% rowSums()
dfr_hits_PW <- amr %>% select(starts_with("PW_dfr")) %>% select(-PW_dfra_any) %>% rowSums()

table(dfr_hits_MYK)
table(dfr_hits_PW)
table(dfr_hits_MYK,dfr_hits_PW)

amr %>% select(starts_with("MYK_dfr"), starts_with("PW_dfr")) %>% select(-PW_dfra_any) %>% group_by(pick(everything()))

amr %>% select(MYK_dfrA1, starts_with("PW_dfr")) %>% select(-PW_dfra_any) %>% group_by(pick(everything())) %>% count()

amr %>% select(MYK_dfrA5, starts_with("PW_dfr")) %>% select(-PW_dfra_any) %>% group_by(pick(everything())) %>% count()

amr %>% select(MYK_dfrA7, starts_with("PW_dfr")) %>% select(-PW_dfra_any) %>% group_by(pick(everything())) %>% count()

amr %>% select(MYK_dfrA14, starts_with("PW_dfr")) %>% select(-PW_dfra_any) %>% group_by(pick(everything())) %>% count()

amr %>% select(MYK_dfrA15, starts_with("PW_dfr")) %>% select(-PW_dfra_any) %>% group_by(pick(everything())) %>% count()

amr %>% select(MYK_dfrA17, starts_with("PW_dfr")) %>% select(-PW_dfra_any) %>% group_by(pick(everything())) %>% count()

amr %>% select(MYK_dfrA18, starts_with("PW_dfr")) %>% select(-PW_dfra_any) %>% group_by(pick(everything())) %>% count()

# presence by PW in those with nothing detected by Mykrobe
amr %>% filter(dfr_hits_MYK==0) %>% select(starts_with("PW_dfr")) %>% select(-PW_dfra_any) %>% group_by(pick(everything())) %>% count()

```

# compare tet calls
``` {r}
# check tet calls 
amr %>% select(starts_with("MYK_tet")) %>% select(-MYK_tetracycline) %>% colSums()

amr %>% select(starts_with("MYK_tet"), starts_with("PW_tet")) %>% select(-MYK_tetracycline, -PW_tetracycline_category) %>% colSums()

amr %>% select(starts_with("MYK_tet"), starts_with("PW_tet")) %>% select(-MYK_tetracycline, -PW_tetracycline_category) %>% group_by(pick(everything())) %>% count()

amr %>% select(MYK_tetA, `PW_tetA(A)`) %>% group_by(pick(everything())) %>% count()
amr %>% select(MYK_tetB, `PW_tetA(B)`) %>% group_by(pick(everything())) %>% count()
amr %>% select(MYK_tetC, `PW_tetA(C)`) %>% group_by(pick(everything())) %>% count()
amr %>% select(MYK_tetD, `PW_tetA(D)`) %>% group_by(pick(everything())) %>% count()

```

# drug-level
``` {r}

amr %>% group_by(MYK_ampicillin, `PW_blaTEM-1D`) %>% count()
amr %>% group_by(MYK_azithromycin, `PW_azith_pred_pheno`) %>% count()
amr %>% group_by(MYK_ceftriaxone, `PW_ESBL_category`) %>% count()
amr %>% group_by(MYK_chloramphenicol, `PW_chloramphenicol_category`) %>% count()
amr %>% group_by(MYK_sulfonamides, `PW_sul_any`) %>% count()
amr %>% group_by(MYK_trimethoprim, `PW_dfra_any`) %>% count()
amr %>% group_by(MYK_tetracycline, `PW_tetracycline_category`) %>% count()

# MDR
amr <- amr %>% mutate(MYK_MDR=if_else(MYK_ampicillin!="S" & MYK_chloramphenicol!="S" & MYK_sulfonamides!="S" & MYK_trimethoprim!="S", 1, 0))

table(amr$MYK_MDR, amr$PW_MDR)

# cipro S/I/R
amr <- amr %>% mutate(MYK_cipSIR=if_else(startsWith(MYK_ciprofloxacin, "I"), "I", "S")) %>% mutate(MYK_cipSIR=if_else(startsWith(MYK_ciprofloxacin, "R"), "R", MYK_cipSIR))

table(amr$MYK_cipSIR, amr$PW_cip_pred_pheno)

# XDR
amr <- amr %>% mutate(MYK_XDR=if_else(MYK_MDR==1 & MYK_ceftriaxone!="S" & MYK_cipSIR=="R", 1, 0))

table(amr$MYK_XDR, amr$PW_XDR)

```

# plasmid inc calls 
``` {r}
#for reference
amr %>% select(starts_with("MYK_Inc")) %>% colnames()

# check IncFIAHI1 calls 
amr %>% mutate(PW_IncFIAHI1=str_count(`PW_Inc Types`, "IncFIA\\(HI1\\)")) %>%
  mutate(PW_IncFIAHI1=if_else(is.na(PW_IncFIAHI1), 0, PW_IncFIAHI1)) %>%
  group_by(MYK_IncFIAHI1, PW_IncFIAHI1) %>% count()

# check IncHI1A calls 
amr %>% mutate(PW_IncHI1A=str_count(`PW_Inc Types`, "IncHI1A")) %>%
  mutate(PW_IncHI1A=if_else(is.na(PW_IncHI1A), 0, PW_IncHI1A)) %>%
  group_by(MYK_IncHI1A, PW_IncHI1A) %>% count()

# check IncHI1BR27 calls 
amr %>% mutate(PW_IncHI1BR27=str_count(`PW_Inc Types`, "IncHI1B\\(R27\\)")) %>%
  mutate(PW_IncHI1BR27=if_else(is.na(PW_IncHI1BR27), 0, PW_IncHI1BR27)) %>%
  group_by(MYK_IncHI1BR27, PW_IncHI1BR27) %>% count()

# IncHI1 calls
amr %>% mutate(PW_IncHI1A=str_count(`PW_Inc Types`, "IncHI1A")) %>%
  mutate(PW_IncHI1A=if_else(is.na(PW_IncHI1A), 0, PW_IncHI1A)) %>%
  mutate(PW_IncHI1BR27=str_count(`PW_Inc Types`, "IncHI1B\\(R27\\)")) %>%
  mutate(PW_IncHI1BR27=if_else(is.na(PW_IncHI1BR27), 0, PW_IncHI1BR27)) %>%
  group_by(MYK_IncHI1BR27, MYK_IncHI1A, PW_IncHI1BR27, PW_IncHI1A) %>% count()

# check IncY calls 
amr %>% mutate(PW_IncY=str_count(`PW_Inc Types`, "IncY")) %>%
  mutate(PW_IncY=if_else(is.na(PW_IncY), 0, PW_IncY)) %>%
  group_by(MYK_IncY, PW_IncY) %>% count()

# check IncX3 calls 
amr %>% mutate(PW_IncX3=str_count(`PW_Inc Types`, "IncX3")) %>%
  mutate(PW_IncX3=if_else(is.na(PW_IncX3), 0, PW_IncX3)) %>%
  group_by(MYK_IncX3, PW_IncX3) %>% count()

amr %>% mutate(PW_IncX1=str_count(`PW_Inc Types`, "IncX1")) %>%
  mutate(PW_IncX1=if_else(is.na(PW_IncX1), 0, PW_IncX1)) %>%
  mutate(PW_IncX3=str_count(`PW_Inc Types`, "IncX3")) %>%
  mutate(PW_IncX3=if_else(is.na(PW_IncX3), 0, PW_IncX3)) %>%
  group_by(MYK_IncX3, PW_IncX3, PW_IncX1) %>% count()

# check IncHI2A calls 
amr %>% mutate(PW_IncHI2A=str_count(`PW_Inc Types`, "IncHI2\\-\\-IncHI2A")) %>%
  mutate(PW_IncHI2A=if_else(is.na(PW_IncHI2A), 0, PW_IncHI2A)) %>%
  group_by(MYK_IncHI2A, PW_IncHI2A) %>% count()

# check IncI1 calls 
amr %>% mutate(PW_IncI1=str_count(`PW_Inc Types`, "IncI1")) %>%
  mutate(PW_IncI1=if_else(is.na(PW_IncI1), 0, PW_IncI1)) %>%
  group_by(MYK_IncI1, PW_IncI1) %>% count()

# check IncL/M calls 
amr %>% mutate(PW_IncL=str_count(`PW_Inc Types`, "IncL")) %>%
  mutate(PW_IncL=if_else(is.na(PW_IncL), 0, PW_IncL)) %>%
  group_by(MYK_IncL_M, PW_IncL) %>% count()

# check IncFIB_pHCM2 calls & PW_IncFIB_K calls, they don't overlap so treat separately
#IncFIB(AP001918) - n=1
#IncFIB(pN55391) - n=1
amr %>% mutate(PW_IncFIB_pHCM2=str_count(`PW_Inc Types`, "IncFIB\\(pHCM2\\)")) %>%
  mutate(PW_IncFIB_pHCM2=if_else(is.na(PW_IncFIB_pHCM2), 0, PW_IncFIB_pHCM2)) %>%
  mutate(PW_IncFIB_K=str_count(`PW_Inc Types`, "IncFIB\\(K\\)")) %>%
  mutate(PW_IncFIB_K=if_else(is.na(PW_IncFIB_K), 0, PW_IncFIB_K)) %>%
  group_by(MYK_IncFIB_pHCM2, MYK_IncFIB_K, PW_IncFIB_pHCM2, PW_IncFIB_K) %>% count()

amr %>% mutate(PW_IncFIB_pHCM2=str_count(`PW_Inc Types`, "IncFIB\\(pHCM2\\)")) %>%
  mutate(PW_IncFIB_pHCM2=if_else(is.na(PW_IncFIB_pHCM2), 0, PW_IncFIB_pHCM2)) %>%
  group_by(MYK_IncFIB_pHCM2, PW_IncFIB_pHCM2) %>% count()

# check IncFIB_K calls 
amr %>% mutate(PW_IncFIB_K=str_count(`PW_Inc Types`, "IncFIB\\(K\\)")) %>%
  mutate(PW_IncFIB_K=if_else(is.na(PW_IncFIB_K), 0, PW_IncFIB_K)) %>%
  group_by(MYK_IncFIB_K, PW_IncFIB_K) %>% count()

# check IncN calls 
amr %>% mutate(PW_IncN=str_count(`PW_Inc Types`, "IncN")) %>%
  mutate(PW_IncN=if_else(is.na(PW_IncN), 0, PW_IncN)) %>%
  group_by(MYK_IncN, PW_IncN) %>% count()

# check other plasmids detected by PW but not included in current release of Mykrobe
amr %>% filter(grepl('IncL', `PW_Inc Types`) |
                 grepl('IncFII', `PW_Inc Types`) |
                 grepl('pN55391', `PW_Inc Types`) |
                 grepl('pCRY', `PW_Inc Types`) |
                 grepl('p0111', `PW_Inc Types`) |
                 grepl('IncQ1', `PW_Inc Types`) |
                 grepl('IncP1', `PW_Inc Types`) |
                 grepl('pXuzhou21', `PW_Inc Types`)) %>% 
  group_by(`PW_Inc Types`) %>% count()

# only plasmid rep with n>1 is p0111 (n=118) but this is not associated with AMR genes (most p0111+ strains do not carry any acquired AMR genes)

#updated in Mykrobe. 
amr %>% filter(grepl('p0111', `PW_Inc Types`)) 
amr %>% filter(grepl('IncX1', `PW_Inc Types`)) 


```

