---
title: "Typhoid Genomics Consortium - Mykrobe - AST"
author: "Danielle Ingle, Kat Holt"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: 
  prettydoc::html_pretty:
    theme: cayman
    toc: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# set working directory to the location of the script
knitr::opts_knit$set(root.dir = getwd())
setwd(dirname(rstudioapi::getSourceEditorContext()$path))

# load packages
library(knitr)
library(tidyverse)
library(dplyr)
library(ggplot2)
library(scales)
library(ComplexUpset)
library(patchwork)
library(readxl)
```

# print session info (package versions)
```{r session_info}

# print versions
sessionInfo()

```


# clean AST analysis
``` {r files}
#include guess max to ensure col types read in correctly
amr_ast <- read_csv("../Supplementary_Table_3.csv", guess_max = min(0, 10000))
```

# make long form
``` {r}
# AST results
ast_long <- amr_ast %>% filter(!is.na(dataset)) %>% 
  select(TGC_sangerlane, dataset, measure, standard, ends_with("Result")) %>% 
  select(-`Ciprofloxacin Result`) %>%
  pivot_longer(cols=ends_with("Result"), names_to="Drug", values_to="Result", names_pattern =  "(.*) Result") %>%
  mutate(Drug=replace(Drug, Drug=="SXT", "Trimethoprim-Sulfamethoxazole"))

# genotypes
geno_long <- amr_ast %>% filter(!is.na(dataset)) %>% 
  select(TGC_sangerlane, ampicillin:tetracycline) %>%
  select(-sulfonamides, -trimethoprim) %>%
  mutate(`ciprofloxacin i`=ciprofloxacin) %>%
  mutate(`ciprofloxacin i`=gsub("I:", "R:", `ciprofloxacin i`)) %>%
  rename(`ciprofloxacin r`=ciprofloxacin) %>%
  mutate(`ciprofloxacin r`=if_else(startsWith(`ciprofloxacin r`,"I"), "S", `ciprofloxacin r`)) %>%
  mutate(ampicillin=if_else(startsWith(ceftriaxone, "R"), "R", ampicillin)) %>%
  pivot_longer(cols=ampicillin:`ciprofloxacin i`, names_to="Drug", values_to="genotype") %>%
  mutate(Drug=str_to_title(Drug)) %>%
  mutate(Drug=replace(Drug, Drug=="Sulfamethoxazole", "Trimethoprim-Sulfamethoxazole"))

# merge geno & pheno
amr_ast_long <- full_join(ast_long, geno_long) %>% 
  filter(!is.na(Result)) %>%
  mutate(genotype_SR=if_else(startsWith(genotype, "R"), "R", genotype)) %>%
  mutate(agree=if_else(genotype_SR==Result, 1, 0)) %>%
  mutate(Drug=replace(Drug, Drug=="Ciprofloxacin I", "Ciprofloxacin (low)")) %>%
  mutate(Drug=replace(Drug, Drug=="Ciprofloxacin R", "Ciprofloxacin (high)")) 

```

# calculate agreement and errors
``` {r}
ast_comparison <- amr_ast_long %>% 
  group_by(Drug, dataset) %>% 
  summarise(n=n(), 
            R=sum(Result=="R"),
            R_percent=R/n,
            agree=sum(agree, na.rm=T), 
            major=sum(Result=="S" & genotype_SR=="R", na.rm=T),
            very_major=sum(Result=="R" & genotype_SR=="S", na.rm=T),
            agree_p=agree/n,
            major_p=major/sum(Result=="S", na.rm=T),
            very_major_p=very_major/sum(Result=="R", na.rm=T)
            )
```

# write full details as supp table
``` {r}
write_csv(file="Supplementary_Table_4.csv",
          ast_comparison %>% 
            rename(Dataset=dataset) %>% 
            mutate(R_percent=paste0(round(R_percent*100,2),"%")) %>%
            mutate(categorical_agreement=paste0(round(agree_p*100,2),"%")) %>%
            mutate(major_error_percent=paste0(round(major_p*100,2),"%")) %>%
            mutate(very_major_error_percent=paste0(round(very_major_p*100,2),"%")) %>%
            select(-agree_p, -major_p, -very_major_p)
            )
```

# Fig 3 - AST prediction plot
``` {r}
ast_comparison %>% 
  mutate(agreement_min=agree_p - sqrt(agree_p*(1-agree_p)/n)*1.96) %>%
  mutate(agreement_max=agree_p + sqrt(agree_p*(1-agree_p)/n)*1.96) %>%
  mutate(agreement_max=replace(agreement_max, agreement_max>1,1)) %>%
  mutate(dataset=factor(dataset,levels=c("CDC", "UKHSA", "SEAP"))) %>%
  mutate(errors=if_else(as.numeric(major_p)>0.03 | as.numeric(very_major_p)>0.03, "flag", "ok")) %>%
  mutate(errors=if_else(is.na(errors), "ok", errors)) %>% # azi UKHSA is NA as 0 called
  mutate(errors2=if_else(as.numeric(major_p)>0.03, "major >3%", "acceptable")) %>%
  mutate(errors2=if_else(as.numeric(very_major_p)>0.015, "very major >1.5%", errors2)) %>%
  mutate(errors2=if_else(is.na(errors2), "acceptable", errors2)) %>% # azi UKHSA is NA as 0 called
  filter(!(dataset=="UKHSA" & Drug=="Ciprofloxacin (high)")) %>%
  ggplot(aes(x=Drug, y=agree_p*100, col=errors2)) + 
  facet_wrap(~dataset) +
  geom_point() + 
  geom_errorbar(aes(ymin=agreement_min*100, ymax=agreement_max*100), width=0.3) + 
  scale_color_manual(values=c("#2b83ba", "#fdae61", "#d7191c")) +
  scale_x_discrete(limits=rev(levels(factor(ast_comparison$Drug)))) +
  coord_flip() + 
  theme_bw() + 
  theme(axis.text.x = element_text(size=8), panel.spacing.x = unit(1, "lines")) +
  labs(y="Categorical agreement (%)", x="", col="Errors")
```

# write plot to PDF & PNG
``` {r}
ggsave("Figure3_categorical_agreement.pdf", width=9, height=3)
ggsave("Figure3_categorical_agreement.png", width=9, height=3, units="in")
```


# numbers for text - cipro
``` {r}
amr_ast %>% filter(dataset=="CDC" & `Ciprofloxacin I Result`=="S" & ciprofloxacin!="S") %>% select(ciprofloxacin, `Ciprofloxacin Measure`)

amr_ast %>% filter(dataset=="CDC" & `Ciprofloxacin I Result`=="S" & ciprofloxacin!="S") %>% select(ciprofloxacin, `Ciprofloxacin Measure`)%>% group_by(ciprofloxacin) %>% count()


amr_ast %>% filter(dataset=="SEAP" & `Ciprofloxacin I Result`=="S" & ciprofloxacin!="S") %>% select(ciprofloxacin, `Ciprofloxacin Measure`)

amr_ast %>% filter(dataset=="SEAP" & `Ciprofloxacin I Result`=="S" & ciprofloxacin!="S") %>% select(ciprofloxacin, `Ciprofloxacin Measure`)%>% group_by(ciprofloxacin) %>% count()

```

# numbers for text - SXT and CIP
``` {r}
amr_ast %>% filter(dataset=="CDC" & `SXT Result`=="R" & `trimethoprim-sulfamethoxazole`=="S") %>% select(`trimethoprim-sulfamethoxazole`, `SXT Measure`, TGC_sangerlane)

amr_ast %>% filter(dataset=="UKHSA" & `SXT Result`=="R" & `trimethoprim-sulfamethoxazole`=="S") %>% select(`trimethoprim-sulfamethoxazole`, trimethoprim, sulfonamides, `SXT Measure`, TGC_sangerlane)


amr_ast %>% filter(dataset=="CDC" & `Ciprofloxacin I Result`=="S" & ciprofloxacin!="S") %>% select(ciprofloxacin, `Ciprofloxacin Measure`)%>% group_by(ciprofloxacin) %>% count()

amr_ast %>% filter(dataset=="CDC" & `Ciprofloxacin R Result`=="R" & !startsWith(ciprofloxacin,"R")) %>% select(ciprofloxacin, `Ciprofloxacin Measure`)


amr_ast %>% filter(dataset=="SEAP" & `Ciprofloxacin I Result`=="S" & ciprofloxacin!="S") %>% select(ciprofloxacin, `Ciprofloxacin Measure`)

amr_ast %>% filter(dataset=="SEAP" & `Ciprofloxacin I Result`=="S" & ciprofloxacin!="S") %>% select(ciprofloxacin, `Ciprofloxacin Measure`)%>% group_by(ciprofloxacin) %>% count()

```

# check discrepant calls vs PW
``` {r}
# read genotype calls from Mykrobe and PW
#need to update to Supp Table 1
amr <- read_csv("../Mykrobe_AMRconcordance_18012024_n11992.csv")

# print the dimensions of the table for reference
dim(data)


cip_ME <- amr_ast %>% filter(`Ciprofloxacin I Result`=="S" & ciprofloxacin!="S") %>% pull(TGC_sangerlane)

amr %>% filter(TGC_sangerlane %in% cip_ME) %>% select(MYK_ciprofloxacin, starts_with("PW_gyr"), starts_with("PW_par"), starts_with("PW_qnr")) %>% view()


cip_VME_cdc <- amr_ast %>% filter(dataset=="CDC" & `Ciprofloxacin R Result`=="R" & !startsWith(ciprofloxacin,"R")) %>% pull(TGC_sangerlane)

amr %>% filter(TGC_sangerlane %in% cip_VME_cdc) %>% select(MYK_ciprofloxacin, starts_with("PW_gyr"), starts_with("PW_par"), starts_with("PW_qnr")) %>% view()


sxt_ME_cdc <- amr_ast %>% filter(dataset=="CDC" & `SXT Result`=="R" & sulfamethoxazole=="S") %>% pull(TGC_sangerlane)

amr %>% filter(TGC_sangerlane %in% sxt_ME_cdc) %>% select(MYK_sulfamethoxazole, starts_with("PW_dfr"), starts_with("PW_sul")) %>% view()

sxt_ME_uk <- amr_ast %>% filter(dataset=="UKHSA" & `SXT Result`=="R" & sulfamethoxazole=="S") %>% pull(TGC_sangerlane)

amr %>% filter(TGC_sangerlane %in% sxt_ME_uk) %>% select(MYK_sulfonamides, MYK_trimethoprim, MYK_sulfamethoxazole, starts_with("PW_dfr"), starts_with("PW_sul")) %>% view()

amr %>% filter(TGC_sangerlane %in% sxt_ME_uk) %>% pull(PW_displayname)



azi_VME_seap <- amr_ast %>% filter(dataset=="SEAP" & `Azithromycin Result`=="R" & !startsWith(azithromycin,"R")) %>% pull(TGC_sangerlane)

amr %>% filter(TGC_sangerlane %in% azi_VME_seap) %>% select(starts_with("PW_acr")) %>% colSums()


cef_VME_seap <- amr_ast %>% filter(dataset=="SEAP" & `Ceftriaxone Result`=="R" & !startsWith(ceftriaxone,"R")) %>% pull(TGC_sangerlane)

amr %>% filter(TGC_sangerlane %in% cef_VME_seap) %>% select(starts_with("PW_bla")) %>% colSums()



cipR_VME_seap <- amr_ast %>% filter(dataset=="SEAP" & `Ciprofloxacin R Result`=="R" & !startsWith(ciprofloxacin,"R")) %>% pull(TGC_sangerlane)

amr %>% filter(TGC_sangerlane %in% cipR_VME_seap) %>% select(starts_with("PW_gyr"), starts_with("PW_par"), starts_with("PW_qnr")) %>% colSums()

amr %>% filter(TGC_sangerlane %in% cipR_VME_seap) %>% group_by(PW_cip_pred_pheno) %>% count()

amr %>% filter(TGC_sangerlane %in% cipR_VME_seap) %>% filter(PW_cip_pred_pheno=="CipR") %>% select(MYK_ciprofloxacin, TGC_sangerlane, PW_displayname, `Data Accession`, starts_with("PW_gyr"), starts_with("PW_par"), starts_with("PW_qnr"))



cipI_VME_seap <- amr_ast %>% filter(dataset=="SEAP" & `Ciprofloxacin I Result`=="R" & ciprofloxacin=="S") %>% pull(TGC_sangerlane)

amr %>% filter(TGC_sangerlane %in% cipI_VME_seap) %>% select(starts_with("PW_gyr"), starts_with("PW_par"), starts_with("PW_qnr")) %>% colSums()

```


# Fig 4 - cip upset plots, CDC/UKHSA
``` {r}

cip_mic <- amr_ast %>% 
  filter(measure =="MIC") %>%
  mutate(cip_MIC=replace(`Ciprofloxacin Measure`, `Ciprofloxacin Measure`=="<0.015", 0.0075)) %>%
  mutate(cip_MIC=replace(cip_MIC, cip_MIC==">8", 16)) %>%
  mutate(cip_MIC=replace(cip_MIC, cip_MIC=="6.40E-02", 0.06)) %>%
  mutate(cip_MIC=replace(cip_MIC, cip_MIC=="1.50E-02", 0.015)) %>%
  mutate(cip_MIC=replace(cip_MIC, cip_MIC==0.12, 0.125)) %>%
  #group_by(cip_MIC) %>% count()
  mutate(GyrA_mutations = str_count(ciprofloxacin, "gyrA")) %>%
  mutate(GyrA83 = str_count(ciprofloxacin, "gyrA_S83")) %>%
  mutate(GyrA87 = str_count(ciprofloxacin, "gyrA_D87")) %>%
  mutate(GyrB464 = str_count(ciprofloxacin, "gyrB_S464")) %>%
  mutate(ParC_mutations = str_count(ciprofloxacin, "parC")) %>% 
  mutate(ParC80 = str_count(ciprofloxacin, "parC_S80")) %>%
  mutate(ParC84 = str_count(ciprofloxacin, "parC_E84")) %>%
  mutate(QRDR_mutations = GyrA_mutations+ParC_mutations+GyrB464) %>%
  mutate(`wt QRDR` = if_else(QRDR_mutations==0, 1, 0)) %>%
  mutate(`qnr gene` = str_count(ciprofloxacin, "qnr")) %>%
  mutate(cip_pheno = if_else(cip_MIC >0.06, "I", "S")) %>%
  mutate(cip_pheno = if_else(cip_MIC >0.5, "R", cip_pheno)) %>%
  mutate(cip_pheno=factor(cip_pheno, levels=c("S","I","R")))


upset_cip_mic_UK <- cip_mic %>%
  filter(dataset=="UKHSA") %>%
  mutate(cip_pheno = if_else(cip_MIC >0.06, "R", "S")) %>%
  upset(name='Genotype', themes=upset_default_themes(text=element_text(size=8)),
      intersect=c( "qnr gene", "GyrB464", "ParC84","ParC80", "GyrA87","GyrA83","wt QRDR"),
      sort_sets=F,
      intersections=list(
        'wt QRDR',
        'GyrB464', 'GyrA83', 'GyrA87',
        c('GyrA83', 'GyrA87'),
        c('GyrA83', 'ParC80'),
        c('GyrA83', 'ParC84'),
        c('GyrA83', 'GyrA87','ParC80'),
        c('wt QRDR', 'qnr gene'),
        c('GyrA83', 'qnr gene'),
        c('GyrB464', 'qnr gene'),
        c('GyrA83', 'GyrA87','ParC80', 'qnr gene')
      ),
      sort_intersections=F,
      set_sizes=(upset_set_size(position='right') +
                    theme(axis.text.x=element_text(angle=45),axis.ticks.x=element_line(),axis.title=element_text(size=8))),
      base_annotations=list( # add stacked barplot
        'Percent of genomes'=list(
        aes=aes(x=intersection, fill=cip_pheno),
          geom=list(
            geom_bar(stat='count', position='fill'), 
            theme(legend.position = "none", axis.title=element_text(size=8)),
            scale_fill_manual(values = c("#cab2d6","#fdbf6f" )),
            scale_y_continuous(labels=scales::percent_format()),
            geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), angle = 45, size=2))
        )
      ), 
      annotations= list( # add violin plot with MIC data
        'MIC measurement (mg/L)'=(
            ggplot(mapping=aes(y=as.numeric(cip_MIC))) +
              geom_hline(aes(yintercept = 0.125), linetype = 1, alpha = 0.6, color = "black")  +
              geom_violin()+ 
              geom_count(aes(colour = cip_pheno))+
              scale_color_manual(values = c("#cab2d6","#fdbf6f" ))+
              scale_y_continuous(trans = log2_trans(), breaks = c(0.0075, 0.015, 0.03, 0.06, 0.125, 0.25, 0.5, 1, 2, 4, 8, 16)) +
              theme(legend.position = "none", axis.title=element_text(size=8)) +
            ggtitle("a) UKHSA")
        )
      )
    ) + patchwork::plot_layout(heights=c(2,1,1)) # relative heights of plotting areas


upset_cip_mic_CDC <- cip_mic %>%
  filter(dataset=="CDC") %>%
  upset(name='Genotype', themes=upset_default_themes(text=element_text(size=8)),
      intersect=c( "qnr gene", "GyrB464", "ParC84","ParC80", "GyrA87","GyrA83","wt QRDR"),
      sort_sets=F,
      intersections=list(
        'wt QRDR',
        'GyrB464', 'GyrA83', 'GyrA87',
        c('GyrA83', 'GyrA87'),
        c('GyrA83', 'ParC80'),
        c('GyrA83', 'ParC84'),
        c('GyrA83', 'GyrA87','ParC80'),
        c('wt QRDR', 'qnr gene'),
        c('GyrA83', 'qnr gene')
      ),
      sort_intersections=F,
      set_sizes=(upset_set_size(position='right') +
                    theme(axis.text.x=element_text(angle=45),axis.ticks.x=element_line(),axis.title=element_text(size=8))),
      guides='over',
      base_annotations=list( # add stacked barplot
        'Percent of genomes'=list(
        aes=aes(x=intersection, fill=cip_pheno),
          geom=list(
            geom_bar(stat='count', position='fill'), 
            theme(legend.position = "none", axis.title=element_text(size=8)),
            scale_fill_manual(values = c("#fdbf6f", "#a6cee3", "#cab2d6")),
            scale_y_continuous(labels=scales::percent_format()),
            geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), angle = 45, size=2))
        )
      ), 
      annotations= list( # add violin plot with MIC data
        'MIC measurement (mg/L)'=(
            ggplot(mapping=aes(y=as.numeric(cip_MIC))) +
              geom_hline(aes(yintercept = 1), linetype = 1, alpha = 0.6, color = "black")  +
              geom_hline(aes(yintercept = 0.125), linetype = 2, alpha = 0.6, color = "black")  +
              geom_violin()+ 
              geom_count(aes(colour = cip_pheno))+
              scale_color_manual(values = c("#fdbf6f", "#a6cee3", "#cab2d6"))+
              scale_y_continuous(trans = log2_trans(), breaks = c(0.0075, 0.015, 0.03, 0.06, 0.125, 0.25, 0.5, 1, 2, 4, 8, 16), limits=c(0.0075, 16)) +
              guides(size = "none") +
              labs(colour = "Phenotype") +
              theme(legend.position = "right", axis.title=element_text(size=8)) +
            ggtitle("b) US CDC")
        )
      )
    ) +
  patchwork::plot_layout(heights=c(2,1,1)) # relative heights of plotting areas

```

# write Fig4 to PDF & PNG
``` {r}
upset_cip_mic_UK | upset_cip_mic_CDC

ggsave("Fig4_Cip_MIC.pdf", width=9, height=5)
ggsave("Fig4_CIP_MIC.png", width=9, height=5, units="in")
```


# Fig 5 - SEAP upset plots

## Fig 5a - azi
``` {r}

azi_dd <- amr_ast %>% 
  filter(dataset =="SEAP") %>%
  filter(!startsWith(`Azithromycin Measure`,"Not")) %>% 
  mutate(`AcrB R717L` = str_count(azithromycin, "acrB_R717L")) %>%
  mutate(`AcrB R717Q` = str_count(azithromycin, "acrB_R717Q")) %>%
  mutate(`wt AcrB` = if_else(`AcrB R717L`==0 & `AcrB R717Q`==0, 1, 0)) %>%
  mutate(azi_pheno = if_else(as.numeric(`Azithromycin Measure`) <=12, "R", "S")) %>%
  mutate(azi_pheno=factor(azi_pheno, levels=c("S","R")))


upset_azi_SEAP <- azi_dd %>%
  upset(name='Genotype', themes=upset_default_themes(text=element_text(size=8)),
      intersect=c("AcrB R717Q","AcrB R717L","wt AcrB"),
      sort_sets=F,
      intersections=list(
        'wt AcrB',
        'AcrB R717L', 
        'AcrB R717Q'
      ),
      sort_intersections=F,
      set_sizes=F,
      guides='over',
      base_annotations=list( # add stacked barplot
        'Percent of genomes'=list(
        aes=aes(x=intersection, fill=azi_pheno),
          geom=list(
            geom_bar(stat='count', position='fill'), 
            theme(legend.position = "none", axis.title=element_text(size=8)),
            scale_fill_manual(values = c("#fdbf6f", "#cab2d6")),
            scale_y_continuous(labels=scales::percent_format()),
            geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), angle = 45, size=2))
        )
      ), 
      annotations= list( # add violin plot with DD data
        'Disk zone measurement (mm)'=(
            ggplot(mapping=aes(y=as.numeric(`Azithromycin Measure`))) +
              geom_hline(aes(yintercept = 12), linetype = 1, alpha = 0.6, color = "black")  +
              geom_violin()+ 
              geom_count(aes(colour = azi_pheno))+
              scale_color_manual(values = c("#fdbf6f", "#cab2d6"))+
              scale_y_continuous(breaks=c(0,10,20,30,40), limits=c(0,40)) +
              theme(legend.position = "none", axis.title=element_text(size=8)) +
            ggtitle("a) Azithromycin")
        )
      )
    ) + patchwork::plot_layout(heights=c(2,1,1)) # relative heights of plotting areas

upset_azi_SEAP
```

## Fig 5a - cef
``` {r}

cef_dd <- amr_ast %>% 
  filter(dataset =="SEAP") %>%
  filter(!startsWith(`Ceftriaxone Measure`,"Not")) %>% 
  mutate(`No ESBL` = if_else(ceftriaxone=="S", 1, 0)) %>%
  mutate(`blaCTX-M-15` = str_count(ceftriaxone, "blaCTX-M-15")) %>%
  mutate(cef_pheno = if_else(as.numeric(`Ceftriaxone Measure`) <=22, "I", "S")) %>%
  mutate(cef_pheno = if_else(as.numeric(`Ceftriaxone Measure`) <=19, "R", cef_pheno)) %>%
  mutate(cef_pheno=factor(cef_pheno, levels=c("S","I","R")))


upset_cef_SEAP <- cef_dd %>%
  upset(name='Genotype', themes=upset_default_themes(text=element_text(size=8)),
      intersect=c("blaCTX-M-15","No ESBL"),
      sort_sets=F,
      intersections=list(
        'No ESBL',
        'blaCTX-M-15'
      ),
      sort_intersections=F,
      set_sizes=F,
      guides='over',
      base_annotations=list( # add stacked barplot
        ' '=list( # using ' ' as the title to suppress the y-axis legend in the multi-panel plot
        aes=aes(x=intersection, fill=cef_pheno),
          geom=list(
            geom_bar(stat='count', position='fill'), 
            theme(legend.position = "none", axis.title=element_text(size=8)),
            scale_fill_manual(values = c("#fdbf6f", "#a6cee3", "#cab2d6")),
            scale_y_continuous(labels=scales::percent_format()),
            geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), angle = 45, size=2))
        )
      ), 
      annotations= list( # add violin plot with DD data
        '   '=(  # using ' ' as the title to suppress the y-axis legend in the multi-panel plot
            ggplot(mapping=aes(y=as.numeric(`Ceftriaxone Measure`))) +
              geom_hline(aes(yintercept = 19), linetype = 1, alpha = 0.6, color = "black")  +
              geom_hline(aes(yintercept = 23), linetype = 2, alpha = 0.6, color = "black")  +
              geom_violin()+ 
              geom_count(aes(colour = cef_pheno))+
              scale_color_manual(values = c("#fdbf6f", "#a6cee3", "#cab2d6"))+
              scale_y_continuous(breaks=c(0,10,20,30,40), limits=c(0,40)) +
              theme(legend.position = "none", axis.title=element_text(size=8)) +
            ggtitle("b) Ceftriaxone")
        )
      )
    ) + patchwork::plot_layout(heights=c(2,1,1)) # relative heights of plotting areas

upset_cef_SEAP
```

## Fig 5a - cipro
``` {r}

cip_dd <- amr_ast %>% 
  filter(dataset =="SEAP") %>%
  filter(!startsWith(`Ciprofloxacin Measure`,"Not")) %>% 
  mutate(GyrA_mutations = str_count(ciprofloxacin, "gyrA")) %>%
  mutate(GyrA83 = str_count(ciprofloxacin, "gyrA_S83")) %>%
  mutate(GyrA87 = str_count(ciprofloxacin, "gyrA_D87")) %>%
  mutate(GyrB464 = str_count(ciprofloxacin, "gyrB_S464")) %>%
  mutate(ParC_mutations = str_count(ciprofloxacin, "parC")) %>% 
  mutate(ParC80 = str_count(ciprofloxacin, "parC_S80")) %>%
  mutate(ParC84 = str_count(ciprofloxacin, "parC_E84")) %>%
  mutate(QRDR_mutations = GyrA_mutations+ParC_mutations+GyrB464) %>%
  mutate(`wt QRDR` = if_else(QRDR_mutations==0, 1, 0)) %>%
  mutate(`qnr gene` = str_count(ciprofloxacin, "qnr")) %>%
  mutate(cip_pheno = if_else(as.numeric(`Ciprofloxacin Measure`) <=30, "I", "S")) %>%
  mutate(cip_pheno = if_else(as.numeric(`Ciprofloxacin Measure`) <=20, "R", cip_pheno)) %>%
  mutate(cip_pheno=factor(cip_pheno, levels=c("S","I","R")))


upset_cip_SEAP <- cip_dd %>%
  upset(name='Genotype', themes=upset_default_themes(text=element_text(size=8)),
      intersect=c( "qnr gene", "GyrB464", "ParC84","ParC80", "GyrA87","GyrA83","wt QRDR"),
      sort_sets=F,
      intersections=list(
        'wt QRDR',
        'GyrB464', 'GyrA83', 'GyrA87',
        c('GyrA83', 'GyrA87'),
        c('GyrA83', 'ParC84'),
        c('GyrA83', 'GyrA87','ParC80'),
        c('GyrA83', 'qnr gene')
      ),
      sort_intersections=F,
      set_sizes=(upset_set_size(position='right') +
                    theme(axis.text.x=element_text(angle=45), axis.ticks.x=element_line(), axis.title=element_text(size=8))),
      guides='over',
      base_annotations=list( # add stacked barplot
        ' '=list( # using ' ' as the title to suppress the y-axis legend in the multi-panel plot
        aes=aes(x=intersection, fill=cip_pheno),
          geom=list(
            geom_bar(stat='count', position='fill'), 
            theme(legend.position = "none", axis.title=element_text(size=8)),
            scale_fill_manual(values = c("#fdbf6f", "#a6cee3", "#cab2d6")),
            scale_y_continuous(labels=scales::percent_format()),
            geom_text(aes(label=..count..), stat="count", position=position_fill(vjust = .5), angle = 45, size=2))
        )
      ), 
      annotations= list( # add violin plot with DD data
        '  '=(  # using '  ' as the title to suppress the y-axis legend in the multi-panel plot
            ggplot(mapping=aes(y=as.numeric(`Ciprofloxacin Measure`))) +
              geom_hline(aes(yintercept = 20), linetype = 1, alpha = 0.6, color = "black")  +
              geom_hline(aes(yintercept = 30), linetype = 2, alpha = 0.6, color = "black")  +
              geom_violin()+ 
              geom_count(aes(colour = cip_pheno))+
              scale_color_manual(values = c("#fdbf6f", "#a6cee3", "#cab2d6"))+
              scale_y_continuous(breaks=c(0,10,20,30,40), limits=c(0,40)) +
              guides(size = "none") +
              labs(colour = "Phenotype") +
              theme(legend.position = "right", axis.title=element_text(size=8)) +
            ggtitle("c) Ciprofloxacin")
        )
      )
    ) + patchwork::plot_layout(heights=c(2,1,1)) # relative heights of plotting areas

upset_cip_SEAP
```
# write Fig5 to PDF & PNG
``` {r}
(upset_azi_SEAP | upset_cef_SEAP | upset_cip_SEAP) + plot_layout(widths=c(1,1,3))

ggsave("Fig5_SEAP_AST.pdf", width=9, height=5)
ggsave("Fig5_SEAP_AST.png", width=9, height=5, units="in")
```
